<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Neo Sans Intel:300,300italic,400,400italic,700,700italic|Times New Roman:300,300italic,400,400italic,700,700italic|Lato:300,300italic,400,400italic,700,700italic|Helvetica:300,300italic,400,400italic,700,700italic|Roboto Mono:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">
<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"itsyunmeng.github.io","root":"/","scheme":"Pisces","version":"7.7.2","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Abstract: This paper shows the preprocessing of nanopore sequencing data, including data splitting (using ont_fast5_api, transfer raw data of fast5 format into fastq format, and merge the results into">
<meta property="og:type" content="article">
<meta property="og:title" content="Nanopore_4.data preprocessing-1.data splitting, basecalling, quality control and filtering">
<meta property="og:url" content="https://itsyunmeng.github.io/2020/06/07/Nanopore4-datapreprocessing/index.html">
<meta property="og:site_name" content="Yunmeng&#39;s Blog">
<meta property="og:description" content="Abstract: This paper shows the preprocessing of nanopore sequencing data, including data splitting (using ont_fast5_api, transfer raw data of fast5 format into fastq format, and merge the results into">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/12.nanoporegenomestitching%20flowchart.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/13.fastq.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/14.phredquality.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/15.qualityref.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results1.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results2.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results3.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results4.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results5.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results6.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results7.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results8.png">
<meta property="og:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results9.png">
<meta property="article:published_time" content="2020-06-07T11:31:03.000Z">
<meta property="article:modified_time" content="2020-06-08T11:15:16.078Z">
<meta property="article:author" content="Yunmeng">
<meta property="article:tag" content="Nanopore">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/12.nanoporegenomestitching%20flowchart.png">

<link rel="canonical" href="https://itsyunmeng.github.io/2020/06/07/Nanopore4-datapreprocessing/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true
  };
</script>

  <title>Nanopore_4.data preprocessing-1.data splitting, basecalling, quality control and filtering | Yunmeng's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <div>
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Yunmeng's Blog</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>


<nav class="site-nav">
  
  <ul id="menu" class="menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-fw fa-home"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-fw fa-user"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-fw fa-tags"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-fw fa-th"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-fw fa-archive"></i>Archives</a>

  </li>
  </ul>

</nav>
</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content">
            

  <div class="posts-expand">
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block " lang="en">
    <link itemprop="mainEntityOfPage" href="https://itsyunmeng.github.io/2020/06/07/Nanopore4-datapreprocessing/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="Yunmeng">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Yunmeng's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Nanopore_4.data preprocessing-1.data splitting, basecalling, quality control and filtering
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-06-07 13:31:03" itemprop="dateCreated datePublished" datetime="2020-06-07T13:31:03+02:00">2020-06-07</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-calendar-check-o"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2020-06-08 13:15:16" itemprop="dateModified" datetime="2020-06-08T13:15:16+02:00">2020-06-08</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Bioinformatics/" itemprop="url" rel="index"><span itemprop="name">Bioinformatics</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>Abstract:</strong> This paper shows the preprocessing of nanopore sequencing data, including data splitting (using ont_fast5_api, transfer raw data of fast5 format into fastq format, and merge the results into a file), basecalling (using guppy), quality control (using nanoplot) and filtering (using filtlong).</p>
<a id="more"></a>
<h1>Nanopore genome stitching flowchart</h1>
<p><img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/12.nanoporegenomestitching%20flowchart.png" alt="12.nanoporegenomestitching flowchart"></p>
<h2 id="Introduction-to-HDF5-FAST5-files">Introduction to HDF5 (FAST5) files</h2>
<ul>
<li>The Hierarchical Data Format version 5 <a href="https://www.neonscience.org/about-hdf5" target="_blank" rel="noopener">(HDF5)</a>, is an open source file format that supports large, complex, heterogeneous data. The <a href="https://medium.com/@shiansu/a-look-at-the-nanopore-fast5-format-f711999e2ff6" target="_blank" rel="noopener">FAST5</a> format is a specification over a HDF5 file, imposing a specific structure over the contents of a HDF5 file. These files are used to store the output of nanopore sequencers. The main data is the “squiggles” that represent pico-amp measurements taken around thousands of times a second at the nanopores.</li>
<li>HDF5 (FAST5) can be open by software <a href="https://www.hdfgroup.org/downloads/hdfview/" target="_blank" rel="noopener">HDFView</a>. Generally, it is not necessary. What we need to do is to idenify the base sequence.</li>
</ul>
<h2 id="Data-splitting">Data splitting</h2>
<h3 id="Introduction-to-ont-fast5-api">Introduction to ont_fast5_api</h3>
<p><a href="https://github.com/nanoporetech/ont_fast5_api" target="_blank" rel="noopener">ont_fast5_api</a> is a simple interface to HDF5 files of the Oxford Nanopore .fast5 file format.<br>
It provides:</p>
<ul>
<li>Concrete implementation of the fast5 file schema using the generic h5py library</li>
<li>Plain-english-named methods to interact with and reflect the fast5 file schema</li>
<li><u>Tools to convert between multi_read and single_read formats</u></li>
<li>Tools to compress/decompress raw data in files</li>
</ul>
<h3 id="Using-ont-fast5-api-to-convert-between-multi-read-and-single-read-formats">Using ont_fast5_api to convert between multi_read and single_read formats</h3>
<ul>
<li>single_to_multi_fast5</li>
</ul>
<figure class="highlight fortran"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">single_to_multi_fast5</span><br><span class="line">    -i, --input_path &lt;(path) folder containing single_read_fast5 files&gt;</span><br><span class="line">    -s, --save_path &lt;(path) to folder <span class="keyword">where</span> multi_read fast5 files will be output&gt;</span><br><span class="line">    [<span class="keyword">optional</span>] -t, --threads &lt;(<span class="built_in">int</span>) <span class="keyword">number</span> of CPU threads to <span class="keyword">use</span>; <span class="keyword">default</span>=<span class="number">1</span>&gt;</span><br><span class="line">    [<span class="keyword">optional</span>] -f, --filename_base &lt;(string) <span class="keyword">name</span> for new multi_read <span class="keyword">file</span>; <span class="keyword">default</span>=<span class="string">"batch"</span> (see note-<span class="number">1</span>)&gt;</span><br><span class="line">    [<span class="keyword">optional</span>] -n, --batch_size &lt;(<span class="built_in">int</span>) <span class="keyword">number</span> of single_reads to <span class="keyword">include</span> <span class="keyword">in</span> each multi_read <span class="keyword">file</span>; <span class="keyword">default</span>=<span class="number">4000</span>&gt;</span><br><span class="line">    [<span class="keyword">optional</span>] --<span class="keyword">recursive</span> &lt;<span class="keyword">if</span> included, recursively search sub-directories for single_read files&gt;</span><br></pre></td></tr></table></figure>
<p>example usage:</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">single_to_multi_fast5 <span class="params">--input_path</span> <span class="string">/data/reads</span> <span class="params">--save_path</span> <span class="string">/data/multi_reads</span></span><br><span class="line">    <span class="params">--filename_base</span> <span class="keyword">batch</span>_output <span class="params">--batch_size</span> 100 <span class="params">--recursive</span></span><br></pre></td></tr></table></figure>
<p>Where /data/reads and/or its subfolders contain single_read .fast5 files. The output will be multi_read fast5 files each containing 100 reads, in the folder: /data/multi_reads with the names: batch_output_0.fast5, batch_output_1.fast5 etc.</p>
<ul>
<li>multi_to_single_fast5</li>
</ul>
<figure class="highlight smali"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">multi_to_single_fast5</span><br><span class="line">    -i, --input_path &lt;(path) folder containing multi_read_fast5 files&gt;</span><br><span class="line">    -s, --save_path &lt;(path) to folder where single_read fast5 files will be output&gt;</span><br><span class="line">    [optional] -t, --threads &lt;(int) number of CPU threads to use; default=1&gt;</span><br><span class="line">    [optional] --recursive &lt;if included, recursively search<span class="built_in"> sub-directories </span>for multi_read files&gt;</span><br></pre></td></tr></table></figure>
<h2 id="Basecalling">Basecalling</h2>
<h3 id="Introduction-to-guppy">Introduction to guppy</h3>
<p>“<a href="https://nanoporetech.com/nanopore-sequencing-data-analysis" target="_blank" rel="noopener">Guppy</a> is a data processing toolkit that contains the Oxford Nanopore Technologies’ <u>basecalling</u> algorithms, and several bioinformatic post-processing features. It is provided as binaries to run on Windows, OS X and Linux platforms, as well as being integrated with MinKNOW, the Oxford Nanopore device control software.”<br>
Early downstream analysis components such as barcoding/demultiplexing, adapter trimming and alignment are contained within Guppy. Furthermore, Guppy now performs modified basecalling (5mC, 6mA and CpG) from the raw signal data, producing an additional FAST5 file of modified base probabilities.<br>
<em>PS: The software is chargeable. if you do sequencing in companies, they will give you fastq files, then there is no need to do this step.</em></p>
<h3 id="Using-guppy-to-do-basecalling">Using guppy to do basecalling</h3>
<p>input file: fast5 file; config file.<br>
output file: fastq file or transformed fast5 file; sequencing_summary.txt; sequence telemetry.js (including the information of temperature/voltage/status of flowcell) and guppy_basecaller_1og-2020-?.log (guppy basecalling software log);</p>
<figure class="highlight sql"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">guppy_basecaller <span class="comment">--help</span></span><br><span class="line">: Guppy Basecalling Software, (C) Oxford Nanopore Technologies, Limited. Version 3.4.5+fb1fbfb</span><br><span class="line">Usage:</span><br><span class="line"><span class="keyword">With</span> config <span class="keyword">file</span>:</span><br><span class="line">  guppy_basecaller -i &lt;<span class="keyword">input</span> <span class="keyword">path</span>&gt; -s &lt;<span class="keyword">save</span> <span class="keyword">path</span>&gt; -c &lt;config <span class="keyword">file</span>&gt; [options]</span><br><span class="line"><span class="keyword">With</span> flowcell <span class="keyword">and</span> kit <span class="keyword">name</span>:</span><br><span class="line">  guppy_basecaller -i &lt;<span class="keyword">input</span> <span class="keyword">path</span>&gt; -s &lt;<span class="keyword">save</span> <span class="keyword">path</span>&gt; <span class="comment">--flowcell &lt;flowcell name&gt;</span></span><br><span class="line">    <span class="comment">--kit &lt;kit name&gt;</span></span><br><span class="line"><span class="keyword">List</span> supported flowcells <span class="keyword">and</span> kits:</span><br><span class="line">  guppy_basecaller <span class="comment">--print_workflows</span></span><br></pre></td></tr></table></figure>
<p>example usage:</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">guppy_basecaller -i fast5_files/ -s output <span class="params">--config</span> <span class="string">/ifs1/Software/biosoft/ont-guppy-cpu/data/dna_r9.4.1_450bps_fast.cfg</span> -r <span class="params">--num_callers</span> 4 <span class="params">--cpu_threads_per_caller</span> 2 <span class="params">--fast5_out</span></span><br><span class="line">cat output/*<span class="string">.fastq</span> &gt;all.fastq</span><br></pre></td></tr></table></figure>
<p>–flowcell: types of flowcell（FLO-MIN106/FLO-MIN107）<br>
–fast5_out: choice of whether to do fast5 output. the fast5 files here are different input file, they contains recognized bases sequencing<br>
–config: two modes, eg. dna_r9.4.1_450bps_fast.cfg or dna_r9.4.1_450bps_hac.cfg. accuracy rate difference of 1~3%.<br>
–num_callers: number of parallel basecallers to create. each num produce a fastq file.<br>
–cpu_threads_per_caller: number of CPU worker threads per basecaller.s<br>
–compress_fastq: Compress fastq output files with gzip<br>
-r: search for input files recursively.<br>
-q: Maximum number of records per fastq file, 0 means use a single file (per worker, per run id).</p>
<blockquote>
<p>Supplementaty：<a href="https://en.wikipedia.org/wiki/FASTQ_format" target="_blank" rel="noopener">FASTQ format</a><br>
FASTQ format is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. <code>.fastq</code> or <code>.fq</code>. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity.<br>
A FASTQ file normally uses four lines per sequence:<br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/13.fastq.png" alt="13.fastq"><br>
Line 1 begins with a ‘@’ character and is followed by a sequence identifier and an optional description (like a FASTA title line).<br>
Line 2 is the raw sequence letters.<br>
Line 3 begins with a ‘+’ character and is optionally followed by the same sequence identifier (and any description) again.<br>
Line 4 encodes the quality values for the sequence in Line 2, and must contain the same number of symbols as letters in the sequence.<br>
Qualiity:<br>
Q20 indicates the error rate is 1%; Q30 indicates the error rate is 0.1%。<br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/14.phredquality.png" alt="14.phredquality"><br>
currently the quality of nanopore is between 7 and 9.<br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/15.qualityref.png" alt="15.qualityref.png"></p>
</blockquote>
<h2 id="Quality-control">Quality control</h2>
<p>Currently the quality of nanopore is between Q7 and Q9 (see the above picture).<br>
Q7 can be used as a standard for data filtering, the average length can be above 20K.</p>
<h3 id="Introduction-to-NanoPlot">Introduction to NanoPlot</h3>
<p><a href="https://github.com/wdecoster/NanoPlot" target="_blank" rel="noopener">NanoPlot</a> can be used for statistical plotting of nanopore data. The input file is in <code>fastq</code> format.<br>
Need to call NanoStat for statistics when plotting.<br>
NanoPlot uses these statistical information for drawing, and eventually generates a web page format file, including histogram of read length, scatter plot of read length and average quality of sequence, etc. At the same time, the software can also plot the statistical file sequencing_summary.txt generated after guppy basecalling.</p>
<h3 id="Using-NanoPlot-to-plot-quality-control-figure">Using NanoPlot to plot quality control figure</h3>
<p>install NanoPlot: better to install NanoPlot on a virtual environment, because different version of NanoPlot needs different python version.<br>
example Usage</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">Nanoplot</span> --<span class="comment">summary</span> <span class="comment">sequencing_summary</span><span class="string">.</span><span class="comment">txt</span> --<span class="comment">loglength</span> <span class="literal">-</span><span class="comment">o</span> <span class="comment">summary</span><span class="literal">-</span><span class="comment">plots</span><span class="literal">-</span><span class="comment">log</span><span class="literal">-</span><span class="comment">transformed</span>  </span><br><span class="line"><span class="comment">NanoPlot</span> <span class="literal">-</span><span class="comment">t</span> <span class="comment">2</span> --<span class="comment">fastq</span> <span class="comment">reads1</span><span class="string">.</span><span class="comment">fastq</span><span class="string">.</span><span class="comment">gz</span> <span class="comment">reads2</span><span class="string">.</span><span class="comment">fastq</span><span class="string">.</span><span class="comment">gz</span> --<span class="comment">maxlength</span> <span class="comment">40000</span> --<span class="comment">plots</span> <span class="comment">hex</span> <span class="comment">dot</span></span><br><span class="line"><span class="comment">NanoPlot</span> <span class="literal">-</span><span class="comment">t</span> <span class="comment">12</span> --<span class="comment">color</span> <span class="comment">yellow</span> --<span class="comment">bam</span> <span class="comment">alignment1</span><span class="string">.</span><span class="comment">bam</span> <span class="comment">alignment2</span><span class="string">.</span><span class="comment">bam</span> <span class="comment">alignment3</span><span class="string">.</span><span class="comment">bam</span> --<span class="comment">downsample</span> <span class="comment">10000</span> <span class="literal">-</span><span class="comment">o</span> <span class="comment">bamplots_downsampled</span></span><br></pre></td></tr></table></figure>
<h2 id="Data-flitering">Data flitering</h2>
<p>Unlike second-generation sequencing, nanopre sequencing is single-end sequencing, each data length is different, and does not include a universal adapter, so the data filtering method is completely different from second-generation sequencing such as illumina.<br>
Filtering data is mainly to filter out some fragments that are too short and those with low sequencing quality. In some cases, if a barcode is included, the head and tail need to be cut.<br>
<em>NOTE： After filtering, the result might be worse. Although some fragments are of low quality, they can be helpful to form a whole frame. If you lose them, you might lose some important information. need to compare the results to determine filtering data or not.</em></p>
<h3 id="Introduction-to-Filtlong">Introduction to Filtlong</h3>
<p><a href="https://github.com/rrwick/Filtlong" target="_blank" rel="noopener">Filtlong</a> is a tool for filtering long reads by quality. It can take a set of long reads and produce a smaller, better subset. It uses both read length (longer is better) and read identity (higher is better) when choosing which reads pass the filter.</p>
<h3 id="Using-Filtlong-to-do-data-filtering">Using Filtlong to do data filtering</h3>
<figure class="highlight vim"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">usage: filtlong &#123;OPTIONS&#125; [input_reads]</span><br><span class="line">Filtlon<span class="variable">g:</span> <span class="keyword">a</span> quality filtering tool <span class="keyword">for</span> Nanopore <span class="built_in">and</span> PacBio reads</span><br><span class="line">positional <span class="keyword">argument</span><span class="variable">s:</span></span><br><span class="line">   input_reads                          <span class="built_in">input</span> long reads <span class="keyword">to</span> <span class="keyword">be</span> filtered</span><br><span class="line">optional <span class="keyword">argument</span><span class="variable">s:</span></span><br><span class="line">   output threshold<span class="variable">s:</span></span><br><span class="line">      -t[<span class="keyword">int</span>], --target_bases [<span class="keyword">int</span>]        keep <span class="keyword">only</span> the best reads <span class="keyword">up</span> <span class="keyword">to</span> this many total bases</span><br><span class="line">      -<span class="keyword">p</span>[float], --keep_percent [float]    keep <span class="keyword">only</span> this percentage of the best reads (measured by</span><br><span class="line">                                           bases)</span><br><span class="line">      --min_length [<span class="keyword">int</span>]                   minimum length threshold</span><br><span class="line">      --min_mean_q [float]                 minimum mean quality threshold</span><br><span class="line">      --min_window_q [float]               minimum window quality threshold</span><br><span class="line">   external references (<span class="keyword">if</span> provided, <span class="keyword">read</span> quality will <span class="keyword">be</span> determined using these instead of from the</span><br><span class="line">   Phred scores):</span><br><span class="line">      -<span class="keyword">a</span>[<span class="keyword">file</span>], --assembly [<span class="keyword">file</span>]          reference assembly in FASTA format</span><br><span class="line">      -<span class="number">1</span>[<span class="keyword">file</span>], --illumina_1 [<span class="keyword">file</span>]        reference Illumina reads in FASTQ format</span><br><span class="line">      -<span class="number">2</span>[<span class="keyword">file</span>], --illumina_2 [<span class="keyword">file</span>]        reference Illumina reads in FASTQ format</span><br><span class="line">   score weights (control the relative contribution of each score <span class="keyword">to</span> the final <span class="keyword">read</span> score):</span><br><span class="line">      --length_weight [float]              weight given <span class="keyword">to</span> the length score (defaul<span class="variable">t:</span> <span class="number">1</span>)</span><br><span class="line">      --mean_q_weight [float]              weight given <span class="keyword">to</span> the mean quality score (defaul<span class="variable">t:</span> <span class="number">1</span>)</span><br><span class="line">      --window_q_weight [float]            weight given <span class="keyword">to</span> the window quality score (defaul<span class="variable">t:</span> <span class="number">1</span>)</span><br><span class="line">   <span class="keyword">read</span> manipulation:</span><br><span class="line">      --trim                               trim non-<span class="keyword">k</span>-mer-matching bases from start/end of reads</span><br><span class="line">      --<span class="keyword">split</span> [<span class="keyword">split</span>]                      <span class="keyword">split</span> reads at this many (<span class="built_in">or</span> more) consecutive</span><br><span class="line">                                           non-<span class="keyword">k</span>-mer-matching bases</span><br><span class="line">   other:</span><br><span class="line">      --window_size [<span class="keyword">int</span>]                  size of sliding window used when measuring window quality</span><br><span class="line">                                           (defaul<span class="variable">t:</span> <span class="number">250</span>)</span><br><span class="line">      --<span class="keyword">verbose</span>                            <span class="keyword">verbose</span> output <span class="keyword">to</span> stderr with info <span class="keyword">for</span> each <span class="keyword">read</span></span><br><span class="line">      --<span class="keyword">version</span>                            <span class="keyword">display</span> the program <span class="keyword">version</span> <span class="built_in">and</span> <span class="keyword">quit</span></span><br><span class="line">   -h, --<span class="keyword">help</span>                           <span class="keyword">display</span> this <span class="keyword">help</span> <span class="keyword">menu</span></span><br></pre></td></tr></table></figure>
<p>example usage</p>
<figure class="highlight angelscript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">filtlong --min_length <span class="number">1000</span> --keep_percent <span class="number">90</span> --target_bases <span class="number">500000000</span> input.fastq.gz | gzip &gt; output.fastq.gz</span><br><span class="line">filtlong <span class="number">-1</span> illumina_1.fastq.gz <span class="number">-2</span> illumina_2.fastq.gz --min_length <span class="number">1000</span> --keep_percent <span class="number">90</span> --target_bases <span class="number">500000000</span> --trim --split <span class="number">500</span> input.fastq.gz | gzip &gt; output.fastq.gz</span><br><span class="line">filtlong <span class="number">-1</span> illumina_1.fastq.gz <span class="number">-2</span> illumina_2.fastq.gz --min_length <span class="number">1000</span> --keep_percent <span class="number">90</span> --target_bases <span class="number">500000000</span> --trim --split <span class="number">500</span> input.fastq.gz | gzip &gt; output.fastq.gz</span><br><span class="line">filtlong <span class="number">-1</span> illumina_1.fastq.gz <span class="number">-2</span> illumina_2.fastq.gz --min_length <span class="number">1000</span> --keep_percent <span class="number">90</span> --target_bases <span class="number">500000000</span> --trim --split <span class="number">1000</span> --length_weight <span class="number">10</span> input.fastq.gz | gzip &gt; output.fastq.gz</span><br><span class="line">filtlong <span class="number">-1</span> illumina_1.fastq.gz <span class="number">-2</span> illumina_2.fastq.gz --min_length <span class="number">1000</span> --keep_percent <span class="number">90</span> --target_bases <span class="number">500000000</span> --trim --split <span class="number">100</span> --mean_q_weight <span class="number">10</span> input.fastq.gz | gzip &gt; output.fastq.gz</span><br></pre></td></tr></table></figure>
<p>filtlong --min_length 500 -min_mean_q 7 …/2.rawdata/minion/all.sra.fastq.gz |  gzip &gt;clean.filtlong.fq.gz<br>
–min_length：Discard any read which is shorter than 500 bp.<br>
–min_mean_q 7：minimum mean quality is 7.<br>
–keep_percent 90：Throw out the worst 10% of reads. This is measured by bp, not by read count. So this option throws out the worst 10% of read bases.<br>
–target_bases 500000000：Remove the worst reads until only 500 Mbp remain, useful for very large read sets. If the input read set is less than 500 Mbp, this setting will have no effect.<br>
–trim: Trim bases from the start and end of reads which do not match a k-mer in the reference. This ensures the each read starts and ends with good sequence.<br>
–split 500: Split reads whenever 500 consequence bases fail to match a k-mer in the reference. This serves to remove very poor parts of reads while keeping the good parts. A lower value will split more aggressively and a higher value will be more conservative. --split 1000: This larger split value makes Filtlong less likely to split a read. I.e. a read has to have a lot of consecutive bad bases before it gets split. This helps to keep the output reads longer.</p>
<h1>Examples</h1>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">mkdir 0607</span><br><span class="line">cp -r <span class="string">/ifs1/TestDatas/namikong/0607/</span>* <span class="string">./</span>   <span class="comment">#my data</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#split and merge fast5 files, using ont_fast5_api</span></span><br><span class="line"><span class="comment">##single to multi</span></span><br><span class="line">single_to_multi_fast5 -i fast5_files/ -s multi -n 4000 <span class="params">--recursive</span></span><br><span class="line"><span class="comment">##multi to single</span></span><br><span class="line">multi_to_single_fast5 -i multi/ -s single <span class="params">--recursive</span> -t 6</span><br><span class="line"></span><br><span class="line"><span class="comment">#basecalling, using guppy</span></span><br><span class="line">guppy_basecaller -i fast5_files/ -s output <span class="params">--config</span> dna_r9.4.1_450bps_fast.cfg -r <span class="params">--num_callers</span> 4 <span class="params">--cpu_threads_per_caller</span> 2 <span class="params">--fast5_out</span></span><br><span class="line">cat output/*<span class="string">.fastq</span> &gt;all.fastq</span><br><span class="line">gzip all.fastq <span class="comment">#save space</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#quality control, using NanoPlot</span></span><br><span class="line">conda create -n nanoplot -y -c bioconda nanoplot</span><br><span class="line">conda activate nanoplot</span><br><span class="line">NanoPlot <span class="params">--fastq</span> all.fastq.gz  -t 12 <span class="params">--plots</span> hex dot pauvre kde -o nanoplot</span><br><span class="line"></span><br><span class="line"><span class="comment">#data filtering, using filtlong</span></span><br><span class="line">filtlong <span class="params">--min_length</span> 500 <span class="params">--min_mean_q</span> 90 all.fastq.gz |  gzip &gt;clean.filtlong.fq.gz <span class="comment">#the read length is above 500 bp, the quality is above 10.</span></span><br></pre></td></tr></table></figure>
<p>results<br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results1.png" alt="1"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results2.png" alt="2"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results3.png" alt="3"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results4.png" alt="4"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results5.png" alt="5"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results6.png" alt="6"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results7.png" alt="7"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results8.png" alt="8"><br>
<img src="https://github.com/ItsYunmeng/MyPostImages/raw/master/Nanopore/16.results9.png" alt="9"></p>
<blockquote>
<p>Tips: put current running project in the background</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ctrl+z <span class="comment">#pause the current projects</span></span><br><span class="line"><span class="built_in">jobs</span> <span class="comment">#see current projects</span></span><br><span class="line"><span class="built_in">bg</span> -1 <span class="comment">#run + id of project</span></span><br><span class="line"><span class="built_in">jobs</span> </span><br></pre></td></tr></table></figure>
</blockquote>
<p>Ref:</p>
<blockquote>
<p><a href="https://denbi-nanopore-training-course.readthedocs.io/en/latest/basecalling/basecalling.html" target="_blank" rel="noopener">Basecalling with Guppy</a></p>
</blockquote>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/Nanopore/" rel="tag"># Nanopore</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/06/06/Nanopore3-installsoftwaresanddownloaddata/" rel="prev" title="Nanopore_3.Install packages and download sequencing data">
      <i class="fa fa-chevron-left"></i> Nanopore_3.Install packages and download sequencing data
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/06/08/Nanopore4-datapreprocessing2/" rel="next" title="Nanopore_4.data preprocessing-2.sequence alignment">
      Nanopore_4.data preprocessing-2.sequence alignment <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  

  </div>


          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">1.</span> <span class="nav-text">Nanopore genome stitching flowchart</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Introduction-to-HDF5-FAST5-files"><span class="nav-number">1.1.</span> <span class="nav-text">Introduction to HDF5 (FAST5) files</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Data-splitting"><span class="nav-number">1.2.</span> <span class="nav-text">Data splitting</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Introduction-to-ont-fast5-api"><span class="nav-number">1.2.1.</span> <span class="nav-text">Introduction to ont_fast5_api</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Using-ont-fast5-api-to-convert-between-multi-read-and-single-read-formats"><span class="nav-number">1.2.2.</span> <span class="nav-text">Using ont_fast5_api to convert between multi_read and single_read formats</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Basecalling"><span class="nav-number">1.3.</span> <span class="nav-text">Basecalling</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Introduction-to-guppy"><span class="nav-number">1.3.1.</span> <span class="nav-text">Introduction to guppy</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Using-guppy-to-do-basecalling"><span class="nav-number">1.3.2.</span> <span class="nav-text">Using guppy to do basecalling</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Quality-control"><span class="nav-number">1.4.</span> <span class="nav-text">Quality control</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Introduction-to-NanoPlot"><span class="nav-number">1.4.1.</span> <span class="nav-text">Introduction to NanoPlot</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Using-NanoPlot-to-plot-quality-control-figure"><span class="nav-number">1.4.2.</span> <span class="nav-text">Using NanoPlot to plot quality control figure</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Data-flitering"><span class="nav-number">1.5.</span> <span class="nav-text">Data flitering</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Introduction-to-Filtlong"><span class="nav-number">1.5.1.</span> <span class="nav-text">Introduction to Filtlong</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Using-Filtlong-to-do-data-filtering"><span class="nav-number">1.5.2.</span> <span class="nav-text">Using Filtlong to do data filtering</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">2.</span> <span class="nav-text">Examples</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Yunmeng"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">Yunmeng</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">67</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">8</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">22</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/ItsYunmeng" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-github"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://twitter.com/Yunmeng_Li" title="Twitter → https:&#x2F;&#x2F;twitter.com&#x2F;Yunmeng_Li" rel="noopener" target="_blank"><i class="fa fa-fw fa-twitter"></i>Twitter</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://www.instagram.com/itsyunmeng" title="Instagram → https:&#x2F;&#x2F;www.instagram.com&#x2F;itsyunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-instagram"></i>Instagram</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yunmengli@hotmail.com" title="E-Mail → mailto:yunmengli@hotmail.com" rel="noopener" target="_blank"><i class="fa fa-fw fa-envelope"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Yunmeng</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> v4.2.0
  </div>
  <span class="post-meta-divider">|</span>
  <div class="theme-info">Theme – <a href="https://pisces.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a> v7.7.2
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  













<script>
if (document.querySelectorAll('pre.mermaid').length) {
  NexT.utils.getScript('//cdn.jsdelivr.net/npm/mermaid@8/dist/mermaid.min.js', () => {
    mermaid.initialize({
      theme    : 'forest',
      logLevel : 3,
      flowchart: { curve     : 'linear' },
      gantt    : { axisFormat: '%m/%d/%Y' },
      sequence : { actorMargin: 50 }
    });
  }, window.mermaid);
}
</script>


  

  

  

</body>
</html>
