<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Neo Sans Intel:300,300italic,400,400italic,700,700italic|Times New Roman:300,300italic,400,400italic,700,700italic|Lato:300,300italic,400,400italic,700,700italic|Helvetica:300,300italic,400,400italic,700,700italic|Roboto Mono:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">
<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"itsyunmeng.github.io","root":"/","scheme":"Pisces","version":"7.7.2","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Abstract: 本文介绍了组装结果的优化原理，纠错软件（pilon和racon）的使用，以及共线性检测（mummer和minimap2）和可视化（dotPlotly）。">
<meta property="og:type" content="article">
<meta property="og:title" content="Nanopore_9.Polishing after stitching 拼接完优化">
<meta property="og:url" content="https://itsyunmeng.github.io/2020/06/13/Nanopore8-polishingafterstitching/index.html">
<meta property="og:site_name" content="Yunmeng&#39;s Blog">
<meta property="og:description" content="Abstract: 本文介绍了组装结果的优化原理，纠错软件（pilon和racon）的使用，以及共线性检测（mummer和minimap2）和可视化（dotPlotly）。">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-06-13T11:23:25.000Z">
<meta property="article:modified_time" content="2020-06-17T19:48:11.295Z">
<meta property="article:author" content="Yunmeng">
<meta property="article:tag" content="Nanopore">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://itsyunmeng.github.io/2020/06/13/Nanopore8-polishingafterstitching/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true
  };
</script>

  <title>Nanopore_9.Polishing after stitching 拼接完优化 | Yunmeng's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <div>
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Yunmeng's Blog</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>


<nav class="site-nav">
  
  <ul id="menu" class="menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-fw fa-home"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-fw fa-user"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-fw fa-tags"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-fw fa-th"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-fw fa-archive"></i>Archives</a>

  </li>
  </ul>

</nav>
</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content">
            

  <div class="posts-expand">
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block " lang="en">
    <link itemprop="mainEntityOfPage" href="https://itsyunmeng.github.io/2020/06/13/Nanopore8-polishingafterstitching/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="Yunmeng">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Yunmeng's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Nanopore_9.Polishing after stitching 拼接完优化
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-06-13 13:23:25" itemprop="dateCreated datePublished" datetime="2020-06-13T13:23:25+02:00">2020-06-13</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-calendar-check-o"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2020-06-17 21:48:11" itemprop="dateModified" datetime="2020-06-17T21:48:11+02:00">2020-06-17</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Bioinformatics/" itemprop="url" rel="index"><span itemprop="name">Bioinformatics</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>Abstract:</strong> 本文介绍了组装结果的优化原理，纠错软件（pilon和racon）的使用，以及共线性检测（mummer和minimap2）和可视化（dotPlotly）。</p>
<a id="more"></a>
<h1>组装结果优化原理</h1>
<h2 id="为什么需要对组装结果进行矫正（polishing">为什么需要对组装结果进行矫正（polishing)?</h2>
<p>由于三代nanopore测序质量比较低，原始数据中存在大量测序错误，即使拼接前进行了纠错，组装结果中仍会存在错误，用长读长或短读长的数据对组装结果进行矫正可以，提高准确率，减少Miscalls，Indels，改善由错装（mis-assemblies)导致的低比对区域。因此，序列拼接完需要对拼接结果进行优化，根据文献报道，经过polish之后，拼接结果与真实基因组（其他测序数据拼接结果）的一致性可以达到99.99%以上。即使组装工具带有纠错功能，仍建议再次进行一轮或多轮的矫正。</p>
<h2 id="为什么需要对组装结果进行矫正（polishing-2">为什么需要对组装结果进行矫正（polishing)?</h2>
<p>这是因为nanopore数据的主要的错误来自于插入与缺失，每次将测序数据与拼接基因组比对能够发现一些错误。下一轮数据与纠错后的序列重新比对，可以发现新的错误，这样经过多轮之后，就可以减少错误。</p>
<blockquote>
<p>组装结果纠错与变异检测、一致性序列的关系：原理是一样的，都是堆叠完找差异；区别是纠错是同一个样本的测序数据；后两者是测序数据和参考序列比。</p>
</blockquote>
<h1>纠错的软件</h1>
<p>常用纠错工具: medaka, pilon, racon, nanopolish等，可以利用三代测序进行纠错，也可以加入二代数据进行纠错。</p>
<h2 id="medaka※">medaka※</h2>
<p><a href="https://github.com/nanoporetech/medaka" target="_blank" rel="noopener">medaka</a>是一个基于叠加序列的一致性序列修正工具，高度推荐使用以获得最佳的一致性准确性。Medaka现可以用于变体识别（variant calling）。使用纳米孔R9.4.1版芯片和最佳的工具，现在你可以进行SNPs识别，获得99%准确率。例如，使用当前的R9.4.1版纳米孔，利用Flip-flop碱基识别器和 Medaka, 测序金黄色葡萄球菌（ <em>S.aureus</em> ）基因组，准确现达到 Q44，其它的小型基因组准确率约 Q40。<br>
只支持三代数据<br>
➢ 由 Oxford Nanopore 开发的开源软件<br>
➢ 仅需使用.fasta 或.fastq 数据<br>
➢ 速度比Nanopolish (输入数据是.fast5格式)快50倍，支持CPU和GPU<br>
➢ 通常在Pomoxis组装后使用<br>
➢ 用FASTQ文档和组装结果作为输入文件<br>
➢ 50X5Mbase基因组用时20分钟<br>
➢ 在Racon基础上，进一步提升了数据准确率<br>
➢ 可针对不同数据进行个性化纠错方法训练<br>
➢ 兼容Linux和MacOS系统<br>
usage:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="selector-id">#conda</span> create -n medaka -c conda-forge -c bioconda -y medaka</span><br><span class="line">conda activate medaka</span><br><span class="line">medaka_consensus -<span class="selector-tag">i</span> 原始纳米孔测序数据 -d 需要纠错的拼接结果 -o 纠错后输出结果文件 -m 芯片类型 -v medaka<span class="selector-class">.vcf</span> -t <span class="number">12</span> &gt;medaka.log</span><br><span class="line">#最终结果是medaka<span class="selector-class">.vcf</span><span class="selector-class">.gz</span> 可能需要运行<span class="number">40</span>min</span><br></pre></td></tr></table></figure>
<p>use cases:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda activate medaka</span><br><span class="line">medaka_consensus -<span class="selector-tag">i</span> /ifs1/TestDatas/namikong/<span class="number">0611</span>/rawdata/SRR8608880<span class="selector-class">.sra</span><span class="selector-class">.fastq</span> -d assembly<span class="selector-class">.fasta</span> -o result -m r941_min_high_g360 -v medaka<span class="selector-class">.vcf</span> -t <span class="number">12</span> &gt;medaka.log</span><br><span class="line">conda deactivate</span><br></pre></td></tr></table></figure>
<h2 id="pilon">pilon</h2>
<p><a href="https://github.com/broadinstitute/pilon" target="_blank" rel="noopener">pilon</a>是由broadinstitute研究所开发的纠错工具，输入原始拼接结果以及原始测序数据比对到拼接结果的bam文件即可。pilon通过比对后的bam文件，可以识别拼接中非一致性的序列，包括单碱基的不同，小的indel，大的indel，后者空位gap，以及错误拼接的区域。输入的bam可以来自于二代测序数据的比对，也可以来自于三代测序数据比对得到的bam，bam文件需要排序并建立索引。<br>
支持二代数据和三代数据</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wget https:<span class="regexp">//gi</span>thub.com<span class="regexp">/broadinstitute/</span>pilon<span class="regexp">/releases/</span>download<span class="regexp">/v1.23/</span>pilon-<span class="number">1.23</span>.jar</span><br><span class="line">java -Xmx16G -jar pilon-<span class="number">1.23</span>.jar</span><br></pre></td></tr></table></figure>
<p>如果要顺利运行程序，要求JAVA &gt; 1.7, 以及根据基因组大小而定的内存，一般而言是1M大小的基因对应1GB的内存。</p>
<blockquote>
<p><a href="https://www.jianshu.com/p/cceeb7d1f413" target="_blank" rel="noopener">三代组装：使用pilon对基因组进行polish</a></p>
</blockquote>
<h3 id="二代数据">二代数据</h3>
<p>illumina数据</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#对拼接结果建立索引</span></span><br><span class="line">bwa index assembly.fasta</span><br><span class="line"></span><br><span class="line"><span class="comment">#illumina比对排序建索引 | bwa, samtools</span></span><br><span class="line">bwa mem -t 24 -R '@RG\tID<span class="function">:foo</span>\tSM<span class="function">:bar</span>:\tPL<span class="function">:ILLUMINA</span>' assembly.fasta <span class="string">/ifs1/TestDatas/nanopore/day2-Assembly/2.rawdata/illumina/SRR8608858.sra_1.fastq.gz</span> <span class="string">/ifs1/TestDatas/nanopore/day2-Assembly/2.rawdata/illumina/SRR8608858.sra_2.fastq.gz</span> &gt;illumina.sam <span class="comment">#二代主流用bwa比对</span></span><br><span class="line">samtools sort -@ 4 -O bam -o illumina.sorted.bam illumina.sam</span><br><span class="line">samtools index illumina.sorted.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用pilon进行纠错</span></span><br><span class="line">java -Xmx32G -jar <span class="string">/ifs1/Software/biosoft/pilon/pilon-1.23.jar</span> <span class="params">--genome</span> assembly.fasta <span class="params">--fix</span> all <span class="params">--changes</span> <span class="params">--frags</span> illumina.sorted.bam <span class="params">--output</span> pillon <span class="params">--outdir</span> result <span class="params">--threads</span> 24 <span class="params">--vcf</span> 2&gt; pilon.log</span><br><span class="line"><span class="comment">##最终结果是pilon.fasta</span></span><br></pre></td></tr></table></figure>
<blockquote>
<p>Usage: bwa mem [options] <idxbase> &lt;in1.fq&gt; [in2.fq]<br>
Usage: pilon --genome genome.fasta [–frags frags.bam] [–jumps jumps.bam] [–unpaired unpaired.bam]<br>
–frags表示输入的是1kb以内的paired-end文库，–jumps表示 大于1k以上的mate pair文库, --bam则是让软件自己猜测</p>
</blockquote>
<h3 id="三代数据">三代数据</h3>
<p>nanopore数据</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#minimap2比对</span></span><br><span class="line">minimap2 -ax map-ont assembly.fasta <span class="string">/ifs1/TestDatas/namikong/0611/rawdata/SRR8608880.sra.fastq</span> &gt;assembly.sam</span><br><span class="line"></span><br><span class="line"><span class="comment">#对bam进行处理</span></span><br><span class="line">samtools sort -@ 4 -O bam -o assembly.sorted.bam assembly.sam</span><br><span class="line">samtools index assembly.sorted.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用pilon进行纠错</span></span><br><span class="line">java -Xmx32G -jar <span class="string">/ifs1/Software/biosoft/pilon/pilon-1.23.jar</span> <span class="params">--genome</span> assembly.fasta <span class="params">--fix</span> all <span class="params">--changes</span> <span class="params">--bam</span> assembly.sorted.bam <span class="params">--output</span> pillon <span class="params">--outdir</span> result <span class="params">--threads</span> 24 <span class="params">--vcf</span> 2&gt; pilon.log</span><br></pre></td></tr></table></figure>
<h2 id="racon">racon</h2>
<p><a href="http://github.com/isovic/racon" target="_blank" rel="noopener">racon</a>是一个基于minimap和miniasm的，构建一致性序列（consensus）的一款软件，也可以用于纠错。既可以用于三代数据也可以用于二代数据的纠错。输入数据需要三个，首先是contig，然后是测序的reads，以及前面二者比对的结果，这个比对结果可以是MHAP、PAF、SAM等三种格式当中的一种即可。数据结果为纠错后的 contig序列。一般racon纠错也可以进行多轮。<br>
支持二代数据（不支持pair end模式）和三代数据，支持CUDA<br>
示例：racon对nanopore数据进行三轮纠错</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#连接原始拼接结果</span></span><br><span class="line">DRAFT=assembly.fasta</span><br><span class="line">READS=<span class="regexp">/ifs1/</span>TestDatas<span class="regexp">/namikong/</span><span class="number">0611</span><span class="regexp">/rawdata/</span>SRR8608880.sra.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment">#minimap2比对</span></span><br><span class="line">minimap2 -t <span class="number">12</span> <span class="variable">$&#123;DRAFT&#125;</span> <span class="variable">$&#123;READS&#125;</span> &gt; round_1.paf</span><br><span class="line"><span class="comment">#racon进行纠错</span></span><br><span class="line"><span class="regexp">/ifs1/</span>Software<span class="regexp">/bin/</span>racon -t <span class="number">24</span> <span class="variable">$&#123;READS&#125;</span> round_1.paf <span class="variable">$&#123;DRAFT&#125;</span> &gt; racon_round1.fasta</span><br><span class="line"></span><br><span class="line"><span class="comment">#第二轮纠错    </span></span><br><span class="line">minimap2 -t <span class="number">12</span> racon_round1.fasta <span class="variable">$&#123;READS&#125;</span> &gt; round_2.paf</span><br><span class="line"><span class="regexp">/ifs1/</span>Software<span class="regexp">/bin/</span>racon -t <span class="number">24</span> <span class="variable">$&#123;READS&#125;</span> round_2.paf racon_round1.fasta&gt; racon_round2.fasta</span><br><span class="line"></span><br><span class="line"><span class="comment">#第三轮纠错</span></span><br><span class="line">minimap2 -t <span class="number">12</span> racon_round2.fasta <span class="variable">$&#123;READS&#125;</span>  &gt; round_3.paf</span><br><span class="line"><span class="regexp">/ifs1/</span>Software<span class="regexp">/bin/</span>racon -t <span class="number">24</span> <span class="variable">$&#123;READS&#125;</span>  round_3.paf racon_round2.fasta&gt; racon_round3.fasta</span><br></pre></td></tr></table></figure>
<blockquote>
<p>预测基因<br>
<code>prodigal -a s1037.faa -d s1037.fnn -i s1037.fna -o s1037.gbk</code><br>
后续可以做功能注释，RNAseq的分析等</p>
</blockquote>
<h1>共线性检测</h1>
<p>纠错前后比较一下，看一不一样（不是特别重要，做不做都行）</p>
<h2 id="利用dotPlotly绘图">利用dotPlotly绘图</h2>
<p><a href="https://github.com/tpoorten/dotPlotly" target="_blank" rel="noopener">dotPlotly</a>：需要依赖 optparse，ggplot2，plotly三个R包。</p>
<h3 id="mummer-dotPlotly">mummer + dotPlotly</h3>
<p>经过比对优化后的结果可以使用mummer进行全局比对，mummer软件包中自带的dnadiff可以对于两条fasta序列直接进行快速比对，找到两条序列之间的差别，比对结果可以使用 mumerplot绘制共线性图。</p>
<figure class="highlight css"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">#利用<span class="selector-tag">mummer</span> +  <span class="selector-tag">dotPlotly</span></span><br><span class="line"><span class="selector-tag">nucmer</span> <span class="selector-tag">--maxmatch</span> <span class="selector-tag">-l</span> 80 <span class="selector-tag">-c</span> 100 <span class="selector-tag">assembly</span><span class="selector-class">.fasta</span> <span class="selector-tag">after_pollish</span><span class="selector-class">.fasta</span> <span class="selector-tag">-p</span> <span class="selector-tag">nucmer</span></span><br><span class="line"><span class="selector-tag">delta-filter</span> <span class="selector-tag">-r</span> <span class="selector-tag">nucmer</span><span class="selector-class">.delta</span> &gt;<span class="selector-tag">filter</span><span class="selector-class">.delta</span></span><br><span class="line"><span class="selector-tag">show-coords</span> <span class="selector-tag">-c</span>  <span class="selector-tag">filter</span><span class="selector-class">.delta</span> &gt;<span class="selector-tag">filter</span><span class="selector-class">.delta</span><span class="selector-class">.coords</span></span><br><span class="line"><span class="selector-tag">mummerCoordsDotPlotly</span><span class="selector-class">.R</span> <span class="selector-tag">-i</span> <span class="selector-tag">filter</span><span class="selector-class">.delta</span><span class="selector-class">.coords</span> <span class="selector-tag">-o</span> <span class="selector-tag">filter</span><span class="selector-class">.delta</span><span class="selector-class">.plot</span> <span class="selector-tag">-m</span> 1000 <span class="selector-tag">q</span> 300000 <span class="selector-tag">-k</span> 10 <span class="selector-tag">-s</span> <span class="selector-tag">-t</span> <span class="selector-tag">-l</span> <span class="selector-tag">-p</span> 12</span><br></pre></td></tr></table></figure>
<p>看生成的filter.delta.plot.png图片</p>
<h3 id="利用Minimap2-dotPlotly">利用Minimap2 + dotPlotly</h3>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">minimap2 -x asm5 assembly<span class="selector-class">.fasta</span> after_pollish<span class="selector-class">.fasta</span> &gt;minimap2<span class="selector-class">.paf</span> pafCoordsDotPlotly<span class="selector-class">.R</span> -<span class="selector-tag">i</span> minimap2<span class="selector-class">.paf</span> -o minimap2<span class="selector-class">.plot</span> -m <span class="number">2000</span> -<span class="selector-tag">q</span> <span class="number">500000</span> k <span class="number">10</span> -s -t -l -<span class="selector-tag">p</span> <span class="number">12</span></span><br></pre></td></tr></table></figure>
<p>看生成的minimap2.plot.html网页文件</p>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/Nanopore/" rel="tag"># Nanopore</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/06/12/Nanopore7-mixedstitching/" rel="prev" title="Nanopore_8.Mixed stitching (of SGS and TGS data)">
      <i class="fa fa-chevron-left"></i> Nanopore_8.Mixed stitching (of SGS and TGS data)
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/06/14/9-variationdetection/" rel="next" title="Nanopore_10.Variation detection 长片段变异检测">
      Nanopore_10.Variation detection 长片段变异检测 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  

  </div>


          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">1.</span> <span class="nav-text">组装结果优化原理</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#为什么需要对组装结果进行矫正（polishing"><span class="nav-number">1.1.</span> <span class="nav-text">为什么需要对组装结果进行矫正（polishing)?</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#为什么需要对组装结果进行矫正（polishing-2"><span class="nav-number">1.2.</span> <span class="nav-text">为什么需要对组装结果进行矫正（polishing)?</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">2.</span> <span class="nav-text">纠错的软件</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#medaka※"><span class="nav-number">2.1.</span> <span class="nav-text">medaka※</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#pilon"><span class="nav-number">2.2.</span> <span class="nav-text">pilon</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#二代数据"><span class="nav-number">2.2.1.</span> <span class="nav-text">二代数据</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#三代数据"><span class="nav-number">2.2.2.</span> <span class="nav-text">三代数据</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#racon"><span class="nav-number">2.3.</span> <span class="nav-text">racon</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">3.</span> <span class="nav-text">共线性检测</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#利用dotPlotly绘图"><span class="nav-number">3.1.</span> <span class="nav-text">利用dotPlotly绘图</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#mummer-dotPlotly"><span class="nav-number">3.1.1.</span> <span class="nav-text">mummer + dotPlotly</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#利用Minimap2-dotPlotly"><span class="nav-number">3.1.2.</span> <span class="nav-text">利用Minimap2 + dotPlotly</span></a></li></ol></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Yunmeng"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">Yunmeng</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">57</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">10</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">14</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/ItsYunmeng" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-github"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://twitter.com/Yunmeng_Li" title="Twitter → https:&#x2F;&#x2F;twitter.com&#x2F;Yunmeng_Li" rel="noopener" target="_blank"><i class="fa fa-fw fa-twitter"></i>Twitter</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://www.instagram.com/itsyunmeng" title="Instagram → https:&#x2F;&#x2F;www.instagram.com&#x2F;itsyunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-instagram"></i>Instagram</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yunmengli@hotmail.com" title="E-Mail → mailto:yunmengli@hotmail.com" rel="noopener" target="_blank"><i class="fa fa-fw fa-envelope"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Yunmeng</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> v4.2.0
  </div>
  <span class="post-meta-divider">|</span>
  <div class="theme-info">Theme – <a href="https://pisces.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a> v7.7.2
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  













<script>
if (document.querySelectorAll('pre.mermaid').length) {
  NexT.utils.getScript('//cdn.jsdelivr.net/npm/mermaid@8/dist/mermaid.min.js', () => {
    mermaid.initialize({
      theme    : 'forest',
      logLevel : 3,
      flowchart: { curve     : 'linear' },
      gantt    : { axisFormat: '%m/%d/%Y' },
      sequence : { actorMargin: 50 }
    });
  }, window.mermaid);
}
</script>


  

  

  

</body>
</html>
