<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Neo Sans Intel:300,300italic,400,400italic,700,700italic|Times New Roman:300,300italic,400,400italic,700,700italic|Lato:300,300italic,400,400italic,700,700italic|Helvetica:300,300italic,400,400italic,700,700italic|Roboto Mono:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">
<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"itsyunmeng.github.io","root":"/","scheme":"Pisces","version":"7.7.2","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="摘要：">
<meta property="og:type" content="article">
<meta property="og:title" content="R包介绍及应用1|phyloseq">
<meta property="og:url" content="https://itsyunmeng.github.io/2020/04/06/Rpackages-phyloseq/index.html">
<meta property="og:site_name" content="Yunmeng&#39;s Blog">
<meta property="og:description" content="摘要：">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://itsyunmeng.github.io/2020/04/06/Rpackages-phyloseq/%EF%BC%9F">
<meta property="article:published_time" content="2020-04-06T08:59:02.000Z">
<meta property="article:modified_time" content="2020-04-06T18:48:50.813Z">
<meta property="article:author" content="Yunmeng">
<meta property="article:tag" content="R.Packages">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://itsyunmeng.github.io/2020/04/06/Rpackages-phyloseq/%EF%BC%9F">

<link rel="canonical" href="https://itsyunmeng.github.io/2020/04/06/Rpackages-phyloseq/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true
  };
</script>

  <title>R包介绍及应用1|phyloseq | Yunmeng's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <div>
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Yunmeng's Blog</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
        <p class="site-subtitle">Time Will Tell</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>


<nav class="site-nav">
  
  <ul id="menu" class="menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-fw fa-home"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-fw fa-user"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-fw fa-tags"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-fw fa-th"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-fw fa-archive"></i>Archives</a>

  </li>
  </ul>

</nav>
</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content">
            

  <div class="posts-expand">
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block " lang="en">
    <link itemprop="mainEntityOfPage" href="https://itsyunmeng.github.io/2020/04/06/Rpackages-phyloseq/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="Yunmeng">
      <meta itemprop="description" content="Junior Researcher | VLIZ | Bacteria in Sea Air">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Yunmeng's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          R包介绍及应用1|phyloseq
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2020-04-06 10:59:02 / Modified: 20:48:50" itemprop="dateCreated datePublished" datetime="2020-04-06T10:59:02+02:00">2020-04-06</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/R/" itemprop="url" rel="index"><span itemprop="name">R</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>摘要：</strong></p>
<a id="more"></a>
<h3 id="phyloseq简介"><a href="#phyloseq简介" class="headerlink" title="phyloseq简介"></a>phyloseq简介</h3><p>微生物群落分析面临许多挑战：整合不同类型的数据和生态学、遗传学、系统发育学、网络分析、可视化以及测试方法。微生物群落数据本身可能来源广泛，如：人体、土壤、海面及海水、污水处理厂、工业设施等; 因此，这些不同的样本类型可能具有非常不同形式和规模的相关数据，这些数据很大程度上取决于实验及其问题。phyloseq包是一个对已经聚类成OTUs后的微生物群落数据（包括：OTU表格、系统发育树、OTU注释信息）进行下游系统发育等分析的综合性工具，集成了微生物群落数据导入、存储、分析和出图等功能。该软件包利用R中的许多工具(vegan, ade4, ape, picante)进行生态学和系统发育分析，同时还使用了先进/灵活的图形系统（ggplot2）以轻松的对系统发育数据绘制发表级别的图表。phyloseq使用S4类的系统将所有相关的系统发育测序数据存储为单个实验级对象，从而更容易共享数据并重现分析。通常，phyloseq寻求促进使用R进行OTU聚类的高通量系统发育测序数据的有效探索和可重复分析。<br>phyloseq的具体功能如下：</p>
<ul>
<li>从流行的Denoising/OTU集群管道（DADA2、UPARSE、QIIME、mothur、BIOM、PyroTagger、RDP等）导入丰度和相关数据</li>
<li>打包了了微生物群落常规分析</li>
<li>支持44种距离方法（UniFrac、Jensen-Shannon等）</li>
<li>排序-&gt;支持多种方法，包括约束排序方法</li>
<li>微生物群落绘图功能：使用ggplot2进行强大、灵活的探索性分析</li>
<li>模块化、可定制的预处理功能，完全支持可重复的工作</li>
<li>基于OTU/样本变量进行数据合并的功能，以及支持手动导入数据</li>
<li>UniFrac距离的多线程并行计算</li>
<li>针对高通量扩增子测序数据的多种测试方法</li>
<li>使用真实发表的数据进行分析和绘图演示</li>
</ul>
<p>phyloseq包在GitHub上是开源的： <a href="https://github.com/joey711/phyloseq" target="_blank" rel="noopener">https://github.com/joey711/phyloseq</a><br>作者尽力在官方文档、引用文献以及其他任何合适的地方引用/注明为此包做出贡献的作者。请随意分享并做出贡献！</p>
<blockquote>
<p>phyloseq: Explore microbiome profiles using R <a href="https://joey711.github.io/phyloseq/" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/</a></p>
</blockquote>
<h3 id="教程"><a href="#教程" class="headerlink" title="教程"></a>教程</h3><h4 id="Phyloseq包安装"><a href="#Phyloseq包安装" class="headerlink" title="Phyloseq包安装"></a>Phyloseq包安装</h4><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">#安装</span><br><span class="line"><span class="function"><span class="title">if</span><span class="params">(!requireNamespace(<span class="string">"BiocManager"</span>)</span></span>)&#123;</span><br><span class="line">  install.packages(<span class="string">"BiocManager"</span>)</span><br><span class="line">&#125;</span><br><span class="line">BiocManager::install(<span class="string">"phyloseq"</span>)</span><br><span class="line">##可选</span><br><span class="line"><span class="function"><span class="title">source</span><span class="params">(<span class="string">'http://bioconductor.org/biocLite.R'</span>)</span></span></span><br><span class="line"><span class="function"><span class="title">biocLite</span><span class="params">(<span class="string">'phyloseq'</span>)</span></span></span><br><span class="line">#版本信息</span><br><span class="line"><span class="function"><span class="title">packageVersion</span><span class="params">(<span class="string">'phyloseq'</span>)</span></span></span><br><span class="line">#加载包</span><br><span class="line"><span class="function"><span class="title">library</span><span class="params">(phyloseq)</span></span></span><br><span class="line"><span class="function"><span class="title">library</span><span class="params">(ggplot2)</span></span></span><br><span class="line">#设置默认绘图主题，ggplot2默认不是这个</span><br><span class="line"><span class="function"><span class="title">theme_set</span><span class="params">(theme_bw()</span></span>)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>ggplot2的theme介绍：<a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" target="_blank" rel="noopener">https://ggplot2.tidyverse.org/reference/ggtheme.html</a><br>phyloseq包安装 <a href="http://joey711.github.io/phyloseq/install.html" target="_blank" rel="noopener">http://joey711.github.io/phyloseq/install.html</a></p>
</blockquote>
<h4 id="导入phyloseq数据"><a href="#导入phyloseq数据" class="headerlink" title="导入phyloseq数据"></a>导入phyloseq数据</h4><p>使用<code>?import</code>便可查看全部数据导入说明及其案例。目前各大主流格式的数据均可导入phyloseq，如BIOM、mothur、PyroTagger、QIIME以及RDP pipeline。<br>下图为phyloseq数据存储和操作的原理示意图。如果你可以将数据导入R中，就可以将其导入phyloseq中。主要包含五种数据类型，分别是以矩阵形式的OTU表otu_table和物种注释表tax_table，数据框形式的mapping文件sample_data，以及进化树文件（read.tree, read.nexus, read_tree）和代表序列文件（DNAStringSet, RNAStringSet, AAStringSet）。<br><img src="%EF%BC%9F" alt="F1"></p>
<h5 id="示例（随机生成的数据）："><a href="#示例（随机生成的数据）：" class="headerlink" title="示例（随机生成的数据）："></a>示例（随机生成的数据）：</h5><figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##otumat</span></span><br><span class="line">otumat = matrix(sample(1:100, 100, replace = <span class="literal">TRUE</span>), nrow = 10, ncol = 10)</span><br><span class="line">View(otumat)</span><br><span class="line">rownames(otumat) &lt;- paste0(<span class="string">"OTU"</span>, 1:nrow(otumat))</span><br><span class="line">colnames(otumat) &lt;- paste0(<span class="string">"Sample"</span>, 1:ncol(otumat))</span><br><span class="line">View(otumat)</span><br><span class="line"><span class="comment">##taxmat</span></span><br><span class="line">taxmat = matrix(sample(letters, 70, replace = <span class="literal">TRUE</span>), nrow = nrow(otumat), ncol = 7)</span><br><span class="line">View(taxmat)</span><br><span class="line">rownames(taxmat) &lt;- rownames(otumat)</span><br><span class="line">colnames(taxmat) &lt;- c(<span class="string">"Domain"</span>, <span class="string">"Phylum"</span>, <span class="string">"Class"</span>, <span class="string">"Order"</span>, <span class="string">"Family"</span>, <span class="string">"Genus"</span>, <span class="string">"Species"</span>)</span><br><span class="line">View(taxmat)</span><br><span class="line">class(otumat)</span><br><span class="line">class(taxmat)</span><br><span class="line"><span class="comment">##combine the otumat and taxmat into a phyloseq object</span></span><br><span class="line">library(<span class="string">"phyloseq"</span>)</span><br><span class="line">OTU = otu_table(otumat, taxa_are_rows = <span class="literal">TRUE</span>)</span><br><span class="line">TAX = tax_table(taxmat)</span><br><span class="line">OTU</span><br><span class="line">TAX</span><br><span class="line"><span class="attribute">physeq</span>=phyloseq(OTU,TAX)</span><br><span class="line">physeq</span><br><span class="line"><span class="comment">#plot Histogram at the Family level</span></span><br><span class="line">plot_bar(physeq, fill = <span class="string">"Family"</span>)</span><br><span class="line"><span class="comment">##sampledata</span></span><br><span class="line">sampledata = sample_data(data.frame(</span><br><span class="line">  Location = sample(LETTERS[1:4], <span class="attribute">size</span>=nsamples(physeq), <span class="attribute">replace</span>=<span class="literal">TRUE</span>),</span><br><span class="line">  Depth = sample(50:1000, <span class="attribute">size</span>=nsamples(physeq), <span class="attribute">replace</span>=<span class="literal">TRUE</span>),</span><br><span class="line">  row.<span class="attribute">names</span>=sample_names(physeq),</span><br><span class="line">  <span class="attribute">stringsAsFactors</span>=<span class="literal">FALSE</span></span><br><span class="line">))</span><br><span class="line">sampledata</span><br><span class="line"><span class="comment">##phylogenetic tree</span></span><br><span class="line">random_tree = rtree(ntaxa(physeq), <span class="attribute">rooted</span>=<span class="literal">TRUE</span>, tip.<span class="attribute">label</span>=taxa_names(physeq))</span><br><span class="line">plot(random_tree)</span><br><span class="line"><span class="comment">##combine the sampledata, phylogenetic tree and physeq into physeq1</span></span><br><span class="line">physeq1 = merge_phyloseq(physeq, sampledata, random_tree)</span><br><span class="line">physeq1</span><br><span class="line"><span class="comment">###or</span></span><br><span class="line">physeq2 = phyloseq(OTU, TAX, sampledata, random_tree)</span><br><span class="line">physeq2</span><br><span class="line"><span class="comment">##Are they identical? YES</span></span><br><span class="line">identical(physeq1,physeq2)</span><br><span class="line"><span class="comment">#使用phyloseq展示进化树和count丰度热图</span></span><br><span class="line">plot_tree(physeq1, <span class="attribute">color</span>=<span class="string">"Location"</span>, label.<span class="attribute">tips</span>=<span class="string">"taxa_names"</span>, <span class="attribute">ladderize</span>=<span class="string">"left"</span>, plot.<span class="attribute">margin</span>=0.3)</span><br><span class="line">plot_tree(physeq1, <span class="attribute">color</span>=<span class="string">"Depth"</span>, <span class="attribute">shape</span>=<span class="string">"Location"</span>, label.<span class="attribute">tips</span>=<span class="string">"taxa_names"</span>, <span class="attribute">ladderize</span>=<span class="string">"right"</span>, plot.<span class="attribute">margin</span>=0.3)</span><br><span class="line">plot_heatmap(physeq1)</span><br><span class="line">plot_heatmap(physeq1, taxa.<span class="attribute">label</span>=<span class="string">"Phylum"</span>)</span><br></pre></td></tr></table></figure>
<h5 id="示例（BIOM格式文件）"><a href="#示例（BIOM格式文件）" class="headerlink" title="示例（BIOM格式文件）"></a>示例（<a href="http://biom-format.org/" target="_blank" rel="noopener">BIOM格式</a>文件）</h5><figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#First, define the file paths. In this case, this will be within the phyloseq package, so we use special features of the `system.file` command to get the paths.</span></span><br><span class="line">rich_dense_biom  = system.file(<span class="string">"extdata"</span>, <span class="string">"rich_dense_otu_table.biom"</span>,  <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">rich_sparse_biom = system.file(<span class="string">"extdata"</span>, <span class="string">"rich_sparse_otu_table.biom"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">min_dense_biom   = system.file(<span class="string">"extdata"</span>, <span class="string">"min_dense_otu_table.biom"</span>,   <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">min_sparse_biom  = system.file(<span class="string">"extdata"</span>, <span class="string">"min_sparse_otu_table.biom"</span>,  <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">treefilename = system.file(<span class="string">"extdata"</span>, <span class="string">"biom-tree.phy"</span>,  <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">refseqfilename = system.file(<span class="string">"extdata"</span>, <span class="string">"biom-refseq.fasta"</span>,  <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#Now that we’ve defined the file paths, let’s use these as argument to the `import_biom` function. Note that the tree and reference sequence files are both suitable for any of the example biom files, which is why we only need one path for each. In practice, you will be specifying a path to a sequence or tree file that matches the rest of your data (include tree tip names and sequence headers)</span></span><br><span class="line">import_biom(rich_dense_biom, treefilename, refseqfilename, <span class="attribute">parseFunction</span>=parse_taxonomy_greengenes)</span><br><span class="line">myData = import_biom(rich_dense_biom, treefilename, refseqfilename, <span class="attribute">parseFunction</span>=parse_taxonomy_greengenes)</span><br><span class="line">myData</span><br><span class="line">sample_data(myData)</span><br><span class="line">plot_tree(myData, <span class="attribute">color</span>=<span class="string">"Genus"</span>, <span class="attribute">shape</span>=<span class="string">"BODY_SITE"</span>, <span class="attribute">size</span>=<span class="string">"abundance"</span>)</span><br><span class="line">plot_richness(myData, <span class="attribute">x</span>=<span class="string">"BODY_SITE"</span>, <span class="attribute">color</span>=<span class="string">"Description"</span>)</span><br><span class="line"></span><br><span class="line">plot_bar(myData, <span class="attribute">fill</span>=<span class="string">"Genus"</span>)</span><br><span class="line">refseq(myData)</span><br></pre></td></tr></table></figure>

<h5 id="示例（qiime输出文件）"><a href="#示例（qiime输出文件）" class="headerlink" title="示例（qiime输出文件）"></a>示例（qiime输出文件）</h5><figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">otufile = system.file(<span class="string">"extdata"</span>, <span class="string">"GP_otu_table_rand_short.txt.gz"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">mapfile = system.file(<span class="string">"extdata"</span>, <span class="string">"master_map.txt"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">trefile = system.file(<span class="string">"extdata"</span>, <span class="string">"GP_tree_rand_short.newick.gz"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">rs_file = system.file(<span class="string">"extdata"</span>, <span class="string">"qiime500-refseq.fasta"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">qiimedata = import_qiime(otufile, mapfile, trefile, rs_file)</span><br><span class="line">qiimedata</span><br><span class="line"></span><br><span class="line">plot_bar(qiimedata, <span class="attribute">x</span>=<span class="string">"SampleType"</span>, <span class="attribute">fill</span>=<span class="string">"Phylum"</span>)</span><br><span class="line">plot_heatmap(qiimedata, sample.<span class="attribute">label</span>=<span class="string">"SampleType"</span>, species.<span class="attribute">label</span>=<span class="string">"Phylum"</span>)</span><br></pre></td></tr></table></figure>
<h5 id="示例（mothur输出文件）"><a href="#示例（mothur输出文件）" class="headerlink" title="示例（mothur输出文件）"></a>示例（mothur输出文件）</h5><figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">mothlist  = system.file(<span class="string">"extdata"</span>, <span class="string">"esophagus.fn.list.gz"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">mothgroup = system.file(<span class="string">"extdata"</span>, <span class="string">"esophagus.good.groups.gz"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">mothtree  = system.file(<span class="string">"extdata"</span>, <span class="string">"esophagus.tree.gz"</span>, <span class="attribute">package</span>=<span class="string">"phyloseq"</span>)</span><br><span class="line">show_mothur_cutoffs(mothlist)</span><br><span class="line">cutoff    = <span class="string">"0.10"</span></span><br><span class="line">x = import_mothur(mothlist, mothgroup, mothtree, cutoff)</span><br><span class="line">x</span><br><span class="line">plot_tree(x, <span class="attribute">color</span>=<span class="string">"samples"</span>)</span><br><span class="line">SDF = data.frame(<span class="attribute">samples</span>=sample_names(x), row.<span class="attribute">names</span>=sample_names(x))</span><br><span class="line">sample_data(x) &lt;- sample_data(SDF)</span><br><span class="line">plot_richness(x)</span><br><span class="line"></span><br><span class="line"><span class="comment">#The class and data in the object returned by import_mothur depends on the arguments. If the first three arguments are provided, then a phyloseq object should be returned containing both a tree and its associated OTU table. If only a list and group file are provided, then an “otu_table” object is returned. Similarly, if only a list and tree file are provided, then only a tree is returned (“phylo” class).</span></span><br><span class="line"><span class="comment">##return just a tree</span></span><br><span class="line">x1 = import_mothur(mothlist, <span class="attribute">mothur_tree_file</span>=mothtree, <span class="attribute">cutoff</span>=<span class="string">"0.10"</span>)</span><br><span class="line">x2 = import_mothur(mothlist, <span class="attribute">mothur_tree_file</span>=mothtree, <span class="attribute">cutoff</span>=<span class="string">"0.08"</span>)</span><br><span class="line">plot(x1)</span><br><span class="line"><span class="comment">##return just an OTU table</span></span><br><span class="line">OTU = import_mothur(mothlist, mothgroup, <span class="attribute">cutoff</span>=<span class="string">"0.08"</span>)</span><br><span class="line">dim(OTU)</span><br><span class="line">head(OTU)</span><br></pre></td></tr></table></figure>
<h5 id="pyrotagger文件不作说明，详见教程。"><a href="#pyrotagger文件不作说明，详见教程。" class="headerlink" title="pyrotagger文件不作说明，详见教程。"></a>pyrotagger文件不作说明，详见教程。</h5><blockquote>
<p>import data: <a href="https://joey711.github.io/phyloseq/import-data.html" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/import-data.html</a></p>
</blockquote>
<h4 id="使用phyloseq的例子"><a href="#使用phyloseq的例子" class="headerlink" title="使用phyloseq的例子"></a>使用phyloseq的例子</h4><figure class="highlight haskell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="class"><span class="keyword">data</span>(<span class="type">GlobalPatterns</span>)</span></span><br><span class="line"><span class="class"><span class="keyword">data</span>(<span class="title">esophagus</span>)</span></span><br><span class="line"><span class="class"><span class="keyword">data</span>(<span class="title">enterotype</span>)</span></span><br><span class="line"><span class="class"><span class="keyword">data</span>(<span class="title">soilrep</span>)</span></span><br><span class="line"></span><br><span class="line">?<span class="type">GlobalPatterns</span></span><br><span class="line"><span class="title">example</span>(enterotype, ask=<span class="type">FALSE</span>)</span><br></pre></td></tr></table></figure>

<blockquote>
<p>example data in phyloseq: <a href="https://joey711.github.io/phyloseq/Example-Data.html" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/Example-Data.html</a></p>
</blockquote>
<h4 id="合并数据"><a href="#合并数据" class="headerlink" title="合并数据"></a>合并数据</h4><ul>
<li><strong>Merging the OTUs or samples in a phyloseq object, based upon a taxonomic or sample variable</strong>: <code>merge_samples()</code> <code>merge_taxa()</code></li>
<li><strong>Merging two or more data objects that come from the same experiment, so that their data becomes part of the same phyloseq object</strong>: <code>merge_phyloseq()</code></li>
</ul>
<h5 id="示例：merge-samples"><a href="#示例：merge-samples" class="headerlink" title="示例：merge_samples"></a>示例：merge_samples</h5><p>如果希望从特定的解释变量中删除重复项之间或样本之间的个体影响，以查看分析会发生什么变化，<code>merge_samples</code>可能非常有用。 使用<code>merge_samples</code>函数，对合并样本的丰度值(abundance values)求和，因此请确保在合并之前进行任何<a href="http://joey711.github.io/phyloseq/preprocess" target="_blank" rel="noopener">预处理</a>以解决测序工作量的差异，否则您将获得测序工作量加权平均值（您可能想要但要请牢记这个）。</p>
<figure class="highlight reasonml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">data(GlobalPatterns)</span><br><span class="line">GP = GlobalPatterns</span><br><span class="line">#数据过滤，保留count大于<span class="number">0</span>的OTU</span><br><span class="line">GP = prune<span class="constructor">_taxa(<span class="params">taxa_sums</span>(GlobalPatterns)</span> &gt; <span class="number">0</span>, GlobalPatterns)</span><br><span class="line">humantypes = c(<span class="string">"Feces"</span>, <span class="string">"Mock"</span>, <span class="string">"Skin"</span>, <span class="string">"Tongue"</span>)</span><br><span class="line">sample<span class="constructor">_data(GP)</span>$human &lt;- get<span class="constructor">_variable(GP, <span class="string">"SampleType"</span>)</span> %<span class="keyword">in</span>% humantypes</span><br><span class="line">#按照sampletype合并样品</span><br><span class="line">mergedGP = merge<span class="constructor">_samples(GP, <span class="string">"SampleType"</span>)</span></span><br><span class="line">SD = merge<span class="constructor">_samples(<span class="params">sample_data</span>(GP)</span>, <span class="string">"SampleType"</span>)</span><br><span class="line">print(SD<span class="literal">[, "S<span class="identifier">ampleType</span>"]</span>)</span><br><span class="line">sample<span class="constructor">_data(<span class="params">mergedGP</span>)</span></span><br><span class="line">print(mergedGP)</span><br><span class="line">sample<span class="constructor">_names(GP)</span></span><br><span class="line">sample<span class="constructor">_names(<span class="params">mergedGP</span>)</span></span><br><span class="line">identical(SD, sample<span class="constructor">_data(<span class="params">mergedGP</span>)</span>)</span><br><span class="line">#如前所述，合并样本的OTU丰度被求和。 让我们自己调查一下这十大最丰富的OTU。</span><br><span class="line">OTUnames10 = names(sort(taxa<span class="constructor">_sums(GP)</span>, TRUE)<span class="literal">[<span class="number">1</span>:<span class="number">10</span>]</span>)</span><br><span class="line">GP10  = prune<span class="constructor">_taxa(OTUnames10,  GP)</span></span><br><span class="line">mGP10 = prune<span class="constructor">_taxa(OTUnames10, <span class="params">mergedGP</span>)</span></span><br><span class="line">ocean_samples = sample<span class="constructor">_names(<span class="params">subset</span>(<span class="params">sample_data</span>(GP)</span>, SampleType==<span class="string">"Ocean"</span>))</span><br><span class="line">print(ocean_samples)</span><br><span class="line">otu<span class="constructor">_table(GP10)</span><span class="literal">[, <span class="identifier">ocean_samples</span>]</span></span><br><span class="line">row<span class="constructor">Sums(<span class="params">otu_table</span>(GP10)</span><span class="literal">[, <span class="identifier">ocean_samples</span>]</span>)</span><br><span class="line">otu<span class="constructor">_table(<span class="params">mGP10</span>)</span><span class="literal">["O<span class="identifier">cean</span>", ]</span></span><br><span class="line">#让我们以图形的方式查看两个丰富度估算摘要图之间的合并。 请注意，我们将忽略自己的建议，并且在合并之前不执行任何预处理。 由于这些是丰富度估算，我们可能是安全的。</span><br><span class="line">plot<span class="constructor">_richness(GP, <span class="string">"human"</span>, <span class="string">"SampleType"</span>, <span class="params">title</span>=<span class="string">"unmerged"</span>)</span></span><br><span class="line">#合并可以做一些奇怪的事情来采样变量。 让我们在绘制之前将这些变量重新添加到sample_data中。</span><br><span class="line">sample<span class="constructor">_data(<span class="params">mergedGP</span>)</span>$SampleType = sample<span class="constructor">_names(<span class="params">mergedGP</span>)</span></span><br><span class="line">sample<span class="constructor">_data(<span class="params">mergedGP</span>)</span>$human = sample<span class="constructor">_names(<span class="params">mergedGP</span>)</span> %<span class="keyword">in</span>% humantypes</span><br><span class="line">plot<span class="constructor">_richness(<span class="params">mergedGP</span>, <span class="string">"human"</span>, <span class="string">"SampleType"</span>, <span class="params">title</span>=<span class="string">"merged"</span>)</span></span><br></pre></td></tr></table></figure>
<p>也许不足为奇，当我们将来自同一环境的大量非重复样本合并在一起时，每种环境的绝对富集度估计值都会增加。 但是，除了示例以外，新的合并图更易于阅读和解释，这可能是使用merge_samples函数的原因之一</p>
<h5 id="示例：merge-taxa"><a href="#示例：merge-taxa" class="headerlink" title="示例：merge_taxa"></a>示例：merge_taxa</h5><p>微生物普查数据集中“噪声”的来源之一是OTU的精细定义，如果我们考虑更高的生物分类等级，该模式就会模糊模式，而这种模式在其他情况下可能是显而易见的。 解决此问题的最佳方法是使用聚集函数<code>tip_glom</code>或<code>tax_glom</code>，它们分别基于系统发生或分类阈值合并相似的OTU。但是，要使这两个功能正常工作，它必须能够合并两个或多个被视为“等效”的OTU。<br>以下是使用<code>merge_taxa</code>的示例，该示例使用树形图显示合并之前和之后，还显示了合并如何不仅影响OTU表，而且还影响具有OTU索引的所有数据组件。<br>我们将使用一个示例数据集，该数据集以前已导入到phyloseq中，变量名为<code>closeps</code>，并保存在RData二进制文件<code>“example-data.RData”</code>中。</p>
<figure class="highlight reasonml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">load(<span class="string">"example-data.RData"</span>)</span><br><span class="line">plot<span class="constructor">_tree(<span class="params">closedps</span>, <span class="params">color</span>=<span class="string">"Treatment"</span>, <span class="params">size</span>=<span class="string">"abundance"</span>, <span class="params">sizebase</span>=2, <span class="params">label</span>.<span class="params">tips</span>=<span class="string">"taxa_names"</span>)</span></span><br><span class="line">#Now <span class="keyword">let</span>’s use merge_taxa <span class="keyword">to</span> merge the first <span class="number">8</span> OTUS <span class="keyword">of</span> closedps into one <span class="keyword">new</span> OTU. By choosing <span class="number">2</span> <span class="keyword">as</span> the optional third argument <span class="keyword">to</span> merge_taxa, we are combining the counts <span class="keyword">of</span> these <span class="number">8</span> OTUs into the index for the second OTU. The default is <span class="keyword">to</span> use the first OTU <span class="keyword">of</span> the second argument.</span><br><span class="line">x1 = merge<span class="constructor">_taxa(<span class="params">closedps</span>, <span class="params">taxa_names</span>(<span class="params">closedps</span>)</span><span class="literal">[<span class="number">3</span>:<span class="number">27</span>]</span>, <span class="number">2</span>)</span><br><span class="line">plot<span class="constructor">_tree(<span class="params">x1</span>, <span class="params">color</span>=<span class="string">"Treatment"</span>, <span class="params">size</span>=<span class="string">"abundance"</span>, <span class="params">sizebase</span>=2, <span class="params">label</span>.<span class="params">tips</span>=<span class="string">"taxa_names"</span>)</span></span><br><span class="line">x1 = merge<span class="constructor">_taxa(<span class="params">closedps</span>, <span class="params">taxa_names</span>(<span class="params">closedps</span>)</span><span class="literal">[<span class="number">3</span>:<span class="number">27</span>]</span>, <span class="number">2</span>)</span><br><span class="line">plot<span class="constructor">_tree(<span class="params">x1</span>, <span class="params">color</span>=<span class="string">"Treatment"</span>, <span class="params">size</span>=<span class="string">"abundance"</span>, <span class="params">sizebase</span>=2, <span class="params">label</span>.<span class="params">tips</span>=<span class="string">"taxa_names"</span>)</span></span><br></pre></td></tr></table></figure>
<h5 id="示例：merge-phyloseq"><a href="#示例：merge-phyloseq" class="headerlink" title="示例：merge_phyloseq"></a>示例：merge_phyloseq</h5><p>As said earlier, merge_phyloseq is a convenience/support tool to help get your data into the right format. Here is an example in which we extract components from an example dataset, and then build them back up to the original form using merge_phyloseq along the way.<br>Let’s split apart the “Global Patterns” example dataset into some components.</p>
<figure class="highlight lisp"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">data(<span class="name">GlobalPatterns</span>)</span><br><span class="line">tree = phy_tree(<span class="name">GlobalPatterns</span>)</span><br><span class="line">tax  = tax_table(<span class="name">GlobalPatterns</span>)</span><br><span class="line">otu  = otu_table(<span class="name">GlobalPatterns</span>)</span><br><span class="line">sam  = sample_data(<span class="name">GlobalPatterns</span>)</span><br><span class="line">otutax = phyloseq(<span class="name">otu</span>, tax)</span><br><span class="line">otutax</span><br></pre></td></tr></table></figure>
<p>As you can see, our new otutax object has just the OTU table and taxonomy table. Now let’s use merge_phyloseq to build up the original GlobalPatterns object, and compare to make sure they are identical. Note how the arguments to merge_phyloseq are a mixture of multi-component (otutax) and single component objects.</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">GP2 = merge_phylose<span class="string">q(otutax, sam, tree)</span></span><br><span class="line">identical(GP2, GlobalPatterns)</span><br></pre></td></tr></table></figure>

<h5 id="示例：更复杂的合并"><a href="#示例：更复杂的合并" class="headerlink" title="示例：更复杂的合并"></a>示例：更复杂的合并</h5><p>The <code>merge_phyloseq</code> function will also work even with more than one multiple-component object.</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">otusamtree = phylose<span class="string">q(otu, sam, tree)</span></span><br><span class="line">GP3 = merge_phylose<span class="string">q(otusamtree, otutax)</span></span><br><span class="line">GP3</span><br><span class="line">identical(GP3, GlobalPatterns)</span><br></pre></td></tr></table></figure>
<p>So this merge appears to have worked. The new object, GP3, looks similar to GlobalPatterns but is not identical. Why? Well, the assumption by merge_phyloseq is that you are atttempting to merge separate sources of abundance data, and so any portion of the OTU tables in the two phyloseq objects that have the same OTU indices are summed together, just like with merge_taxa earlier.<br>This example scenario was illustrative, but hopefully rare in practice. Nevertheless, just in case, an easy fix would be to extract the unique component of otutax and provide it to merge_phyloseq, instead of the entire phyloseq object. This amounts to a small intuitive modification to the previous merge_phyloseq command:</p>
<figure class="highlight lisp"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">GP4 = merge_phyloseq(<span class="name">otusamtree</span>, tax_table(<span class="name">otutax</span>))</span><br><span class="line">GP4</span><br><span class="line">identical(<span class="name">GP4</span>, GlobalPatterns)</span><br></pre></td></tr></table></figure>

<blockquote>
<p>merge data: <a href="https://joey711.github.io/phyloseq/merge.html" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/merge.html</a></p>
</blockquote>
<h4 id="访问数据和数据预处理"><a href="#访问数据和数据预处理" class="headerlink" title="访问数据和数据预处理"></a>访问数据和数据预处理</h4><h5 id="Accessors"><a href="#Accessors" class="headerlink" title="Accessors"></a>Accessors</h5><p>Components of a phyloseq object, like the OTU Table, can be accessed by special accessor functions, or <code>accessors</code>, which return specific information about phylogenetic sequencing data, if present. These accessor functions are available for direct interaction by users and dependent functions/packages</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">##访问数据及其数据处理</span><br><span class="line"><span class="function"><span class="title">data</span><span class="params">(<span class="string">"GlobalPatterns"</span>)</span></span></span><br><span class="line">GlobalPatterns</span><br><span class="line">#查看OTU数量</span><br><span class="line"><span class="function"><span class="title">ntaxa</span><span class="params">(GlobalPatterns)</span></span></span><br><span class="line">#查看样品数量</span><br><span class="line"><span class="function"><span class="title">nsamples</span><span class="params">(GlobalPatterns)</span></span></span><br><span class="line">#样品名查看，此处查看前五个样品名称</span><br><span class="line"><span class="function"><span class="title">sample_names</span><span class="params">(GlobalPatterns)</span></span>[<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line">#查看tax的分类等级信息</span><br><span class="line"><span class="function"><span class="title">rank_names</span><span class="params">(GlobalPatterns)</span></span></span><br><span class="line">#查看mapping文件表头（样品信息文件列名）</span><br><span class="line"><span class="function"><span class="title">sample_variables</span><span class="params">(GlobalPatterns)</span></span></span><br><span class="line">#查看部分OTU表格矩阵</span><br><span class="line"><span class="function"><span class="title">otu_table</span><span class="params">(GlobalPatterns)</span></span>[<span class="number">1</span>:<span class="number">5</span>, <span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line">#查看部分注释（tax）文件矩阵</span><br><span class="line"><span class="function"><span class="title">tax_table</span><span class="params">(GlobalPatterns)</span></span>[<span class="number">1</span>:<span class="number">5</span>, <span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">#进化树查看，注意不是可视化</span><br><span class="line"><span class="function"><span class="title">phy_tree</span><span class="params">(GlobalPatterns)</span></span></span><br><span class="line">#查看OTU名称，此处查看前十个</span><br><span class="line"><span class="function"><span class="title">taxa_names</span><span class="params">(GlobalPatterns)</span></span>[<span class="number">1</span>:<span class="number">10</span>]</span><br><span class="line"></span><br><span class="line">#按照丰度提取前十个OTU，并可视化进化树</span><br><span class="line">myTaxa = names(sort(taxa_sums(GlobalPatterns), decreasing = TRUE)[<span class="number">1</span>:<span class="number">10</span>])</span><br><span class="line">ex1 = prune_taxa(myTaxa, GlobalPatterns)</span><br><span class="line"><span class="function"><span class="title">plot</span><span class="params">(phy_tree(ex1)</span></span>, show<span class="selector-class">.node</span><span class="selector-class">.label</span> = TRUE)</span><br><span class="line"></span><br><span class="line"><span class="function"><span class="title">plot_tree</span><span class="params">(ex1, color = <span class="string">"SampleType"</span>, label.tips = <span class="string">"Phylum"</span>, ladderize = <span class="string">"left"</span>, justify = <span class="string">"left"</span> , size = <span class="string">"Abundance"</span>)</span></span></span><br></pre></td></tr></table></figure>
<h5 id="preprocessing"><a href="#preprocessing" class="headerlink" title="preprocessing"></a>preprocessing</h5><figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#转化OTUcount数为相对丰度</span></span><br><span class="line">GPr  = transform_sample_counts(GlobalPatterns, function(x) x / sum(x) )</span><br><span class="line">otu_table(GPr)[1:5][1:5]</span><br><span class="line"><span class="comment">#提取丰度大于十万份之一的OTU</span></span><br><span class="line">GPfr = filter_taxa(GPr, function(x) mean(x) &gt; 1e-5, <span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment">#提取指定分类的OTU</span></span><br><span class="line">GP.chl = subset_taxa(GlobalPatterns, <span class="attribute">Phylum</span>=="Chlamydiae")</span><br><span class="line"><span class="comment">#提取总count数量大于20的样品</span></span><br><span class="line">GP.chl = prune_samples(sample_sums(GP.chl)&gt;=20, GP.chl)</span><br><span class="line"><span class="comment">#合并指定OTU： taxa_names(GP.chl)[1:5]为一个OTU</span></span><br><span class="line">GP.chl.merged = merge_taxa(GP.chl, taxa_names(GP.chl)[1:5])</span><br><span class="line"><span class="comment"># gpsfbg = tax_glom(gpsfb, "Family")</span></span><br><span class="line"><span class="comment"># plot_tree(gpsfbg, color="SampleType", shape="Class", size="abundance")</span></span><br><span class="line"></span><br><span class="line">transform_sample_counts(GP.chl, function(OTU) OTU/sum(OTU) )</span><br><span class="line"><span class="comment">#去除至少20％样本中未见过3次以上的OTU,注意编写函数的形式</span></span><br><span class="line">GP = filter_taxa(GlobalPatterns, function(x) sum(x &gt; 3) &gt; (0.2*length(x)), <span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment">#对样品分组文件mapping添加一列</span></span><br><span class="line">sample_data(GP)<span class="variable">$human</span> = factor( get_variable(GP, <span class="string">"SampleType"</span>) %<span class="keyword">in</span>% c(<span class="string">"Feces"</span>, <span class="string">"Mock"</span>, <span class="string">"Skin"</span>, <span class="string">"Tongue"</span>) )</span><br><span class="line"></span><br><span class="line"><span class="comment">#测序深度进行统一</span></span><br><span class="line">total = median(sample_sums(GP))</span><br><span class="line">standf = function(x, <span class="attribute">t</span>=total) round(t * (x / sum(x)))</span><br><span class="line">gps = transform_sample_counts(GP, standf)</span><br><span class="line">sample_sums(gps)</span><br><span class="line"><span class="comment">#变异系数(3.0)过滤OTU，减少一些意外OTU误差</span></span><br><span class="line">gpsf = filter_taxa(gps, function(x) sd(x)/mean(x) &gt; 3.0, <span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment">#提取指定门类OTU</span></span><br><span class="line">gpsfb = subset_taxa(gpsf, <span class="attribute">Phylum</span>=="Bacteroidetes")</span><br><span class="line"></span><br><span class="line">title = <span class="string">"plot_bar; Bacteroidetes-only"</span></span><br><span class="line">plot_bar(gpsfb, <span class="string">"SampleType"</span>, <span class="string">"Abundance"</span>, <span class="attribute">title</span>=title)</span><br><span class="line">plot_bar(gpsfb, <span class="string">"SampleType"</span>, <span class="string">"Abundance"</span>, <span class="string">"Family"</span>, <span class="attribute">title</span>=title)</span><br><span class="line">plot_bar(gpsfb, <span class="string">"Family"</span>, <span class="string">"Abundance"</span>, <span class="string">"Family"</span>,</span><br><span class="line">         <span class="attribute">title</span>=title, <span class="attribute">facet_grid</span>=<span class="string">"SampleType~."</span>)</span><br></pre></td></tr></table></figure>

<blockquote>
<p>functions for accessing and (pre)processing data: <a href="https://joey711.github.io/phyloseq/preprocess.html" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/preprocess.html</a></p>
</blockquote>
<h4 id="基于微生物群落数据距离的计算"><a href="#基于微生物群落数据距离的计算" class="headerlink" title="基于微生物群落数据距离的计算"></a>基于微生物群落数据距离的计算</h4><p><code>distanceMethodList</code>：查看所有的方法<br><code>？distance</code>：查看使用方法，</p>
<ul>
<li>usage：<code>distance(physeq, method, type = &quot;samples&quot;, ...)</code></li>
</ul>
<p>示例（使用<code>Enterotypes</code>数据集）：</p>
<figure class="highlight nix"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##基于phyloseq的微生物群落聚类的计算</span></span><br><span class="line">library(<span class="string">"plyr"</span>); packageVersion(<span class="string">"plyr"</span>)</span><br><span class="line">data(enterotype)</span><br><span class="line"><span class="comment">#删除包含所有未分配序列（“-1”）的OTU</span></span><br><span class="line">enterotype &lt;- subset_taxa(enterotype, Genus != <span class="string">"-1"</span>)</span><br><span class="line"><span class="comment">#我们进行查看发现有很多的距离算法</span></span><br><span class="line">dist_methods &lt;- unlist(distanceMethodList)</span><br><span class="line">print(dist_methods)</span><br><span class="line"><span class="comment">#删除需要tree的距离算法`UniFrac1` `UniFrac2` `DPCoA`</span></span><br><span class="line">dist_methods &lt;- dist_methods[-(<span class="number">1</span>:<span class="number">3</span>)]</span><br><span class="line"><span class="comment">#删除需要用户自定义的距离算法`designdist`</span></span><br><span class="line"><span class="attr">dist_methods</span> = dist_methods[-which(<span class="attr">dist_methods=="ANY")]</span></span><br><span class="line"><span class="comment">#loop through each distance method, save each plot to a list, called `plist`</span></span><br><span class="line">plist &lt;- vector(<span class="string">"list"</span>, length(dist_methods))</span><br><span class="line">names(plist) = dist_methods</span><br><span class="line">for( i <span class="keyword">in</span> dist_methods )&#123;</span><br><span class="line">  <span class="comment"># Calculate distance matrix</span></span><br><span class="line">  iDist &lt;- distance(enterotype, <span class="attr">method=i)</span></span><br><span class="line">  <span class="comment"># Calculate ordination</span></span><br><span class="line">  iMDS  &lt;- ordinate(enterotype, <span class="string">"MDS"</span>, <span class="attr">distance=iDist)</span></span><br><span class="line">  <span class="comment">## Make plot</span></span><br><span class="line">  <span class="comment"># Don't carry over previous plot (if error, p will be blank)</span></span><br><span class="line">  p &lt;- NULL</span><br><span class="line">  <span class="comment"># Create plot, store as temp variable, p</span></span><br><span class="line">  p &lt;- plot_ordination(enterotype, iMDS, <span class="attr">color="SeqTech",</span> <span class="attr">shape="Enterotype")</span></span><br><span class="line">  <span class="comment"># Add title to each plot</span></span><br><span class="line">  p &lt;- p + ggtitle(paste(<span class="string">"MDS using distance method "</span>, i, <span class="attr">sep=""))</span></span><br><span class="line">  <span class="comment"># Save the graphic to file.</span></span><br><span class="line">  plist[[i]] = p</span><br><span class="line">&#125;</span><br><span class="line"><span class="comment">#合并结果</span></span><br><span class="line"><span class="comment">##Shade according to sequencing technology</span></span><br><span class="line"><span class="attr">df</span> = ldply(plist, function(x) x$data)</span><br><span class="line">names(df)[<span class="number">1</span>] &lt;- <span class="string">"distance"</span></span><br><span class="line"><span class="attr">p</span> = ggplot(df, aes(Axis.<span class="number">1</span>, Axis.<span class="number">2</span>, <span class="attr">color=SeqTech,</span> <span class="attr">shape=Enterotype))</span></span><br><span class="line"><span class="attr">p</span> = p + geom_point(<span class="attr">size=3,</span> <span class="attr">alpha=0.5)</span></span><br><span class="line"><span class="attr">p</span> = p + facet_wrap(~distance, <span class="attr">scales="free")</span></span><br><span class="line"><span class="attr">p</span> = p + ggtitle(<span class="string">"MDS on various distance metrics for Enterotype dataset"</span>)</span><br><span class="line">p</span><br><span class="line"><span class="comment">#Shade according to assigned enterotype</span></span><br><span class="line"><span class="attr">df</span> = ldply(plist, function(x) x$data)</span><br><span class="line">names(df)[<span class="number">1</span>] &lt;- <span class="string">"distance"</span></span><br><span class="line"><span class="attr">p</span> = ggplot(df, aes(Axis.<span class="number">1</span>, Axis.<span class="number">2</span>, <span class="attr">color=Enterotype,</span> <span class="attr">shape=SeqTech))</span></span><br><span class="line"><span class="attr">p</span> = p + geom_point(<span class="attr">size=3,</span> <span class="attr">alpha=0.5)</span></span><br><span class="line"><span class="attr">p</span> = p + facet_wrap(~distance, <span class="attr">scales="free")</span></span><br><span class="line"><span class="attr">p</span> = p + ggtitle(<span class="string">"MDS on various distance metrics for Enterotype dataset"</span>)</span><br><span class="line">p</span><br><span class="line"><span class="comment">#比较结果</span></span><br><span class="line"><span class="comment">##jsd（Jensen-Shannon Divergence）距离查看效果</span></span><br><span class="line">print(plist[[<span class="string">"jsd"</span>]])</span><br><span class="line"><span class="comment">##jaccard距离查看效果</span></span><br><span class="line">print(plist[[<span class="string">"jaccard"</span>]])</span><br><span class="line"><span class="comment">##Bray-Curtis距离查看效果</span></span><br><span class="line">print(plist[[<span class="string">"bray"</span>]])</span><br><span class="line"><span class="comment">##Gower距离查看效果</span></span><br><span class="line">print(plist[[<span class="string">"gower"</span>]])</span><br><span class="line"><span class="comment">##w距离查看效果</span></span><br><span class="line">print(plist[[<span class="string">"w"</span>]])</span><br></pre></td></tr></table></figure>

<blockquote>
<p>the distance function in phyloseq: <a href="https://joey711.github.io/phyloseq/distance.html" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/distance.html</a></p>
</blockquote>
<h4 id="差异统计"><a href="#差异统计" class="headerlink" title="差异统计"></a>差异统计</h4><figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br></pre></td><td class="code"><pre><span class="line">library(<span class="string">"phyloseq"</span>); packageVersion(<span class="string">"phyloseq"</span>)</span><br><span class="line">library(<span class="string">"cluster"</span>); packageVersion(<span class="string">"cluster"</span>)</span><br><span class="line">library(<span class="string">"ggplot2"</span>); packageVersion(<span class="string">"ggplot2"</span>)</span><br><span class="line">theme_set(theme_bw())</span><br><span class="line"><span class="comment">#Load data</span></span><br><span class="line">data(enterotype)</span><br><span class="line"><span class="comment">#排序ordination</span></span><br><span class="line"><span class="comment">##jsd距离进行MDS排序</span></span><br><span class="line">exord = ordinate(enterotype, <span class="attribute">method</span>=<span class="string">"MDS"</span>, <span class="attribute">distance</span>=<span class="string">"jsd"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#conpute gap statistic</span></span><br><span class="line">pam1 = function(x, k)&#123;list(cluster = pam(x,k, cluster.<span class="attribute">only</span>=<span class="literal">TRUE</span>))&#125;</span><br><span class="line">x = phyloseq:::scores.pcoa(exord, <span class="attribute">display</span>=<span class="string">"sites"</span>)</span><br><span class="line"><span class="comment"># gskmn = clusGap(x[, 1:2], FUN=kmeans, nstart=20, K.max = 6, B = 500)</span></span><br><span class="line">gskmn = clusGap(x[, 1:2], <span class="attribute">FUN</span>=pam1, K.max = 6, B = 50)</span><br><span class="line">gskmn</span><br><span class="line"></span><br><span class="line"><span class="comment">#Pretty straightforward. In case it is useful to see, this is what a wrapper-function might look like to “add-on” code to phyloseq.</span></span><br><span class="line">gap_statistic_ordination = function(ord, FUNcluster, <span class="attribute">type</span>=<span class="string">"sites"</span>, K.<span class="attribute">max</span>=6, <span class="attribute">axes</span>=c(1:2), <span class="attribute">B</span>=500, <span class="attribute">verbose</span>=interactive(), <span class="built_in">..</span>.)&#123;</span><br><span class="line">  require(<span class="string">"cluster"</span>)</span><br><span class="line">  #   <span class="keyword">If</span> <span class="string">"pam1"</span> was chosen, use this internally defined call <span class="keyword">to</span> pam</span><br><span class="line">  <span class="keyword">if</span>(FUNcluster == <span class="string">"pam1"</span>)&#123;</span><br><span class="line">    FUNcluster = function(x,k) list(cluster = pam(x, k, cluster.<span class="attribute">only</span>=<span class="literal">TRUE</span>))     </span><br><span class="line">  &#125;</span><br><span class="line">  # Use the scores function <span class="keyword">to</span> <span class="builtin-name">get</span> the ordination coordinates</span><br><span class="line">  x = phyloseq:::scores.pcoa(ord, <span class="attribute">display</span>=type)</span><br><span class="line">  #   <span class="keyword">If</span> axes <span class="keyword">not</span> explicitly defined (<span class="literal">NULL</span>), then use all of them</span><br><span class="line">  <span class="keyword">if</span>(is.<span class="literal">null</span>(axes))&#123;axes = 1:ncol(x)&#125;</span><br><span class="line">  #   Finally, perform, <span class="keyword">and</span> return, the gap statistic calculation using cluster::clusGap  </span><br><span class="line">  clusGap(x[, axes], <span class="attribute">FUN</span>=FUNcluster, K.<span class="attribute">max</span>=K.max, <span class="attribute">B</span>=B, <span class="attribute">verbose</span>=verbose, <span class="built_in">..</span>.)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">#plot results</span></span><br><span class="line">plot_clusgap = function(clusgap, <span class="attribute">title</span>=<span class="string">"Gap Statistic calculation results"</span>)&#123;</span><br><span class="line">  require(<span class="string">"ggplot2"</span>)</span><br><span class="line">  gstab = data.frame(clusgap<span class="variable">$Tab</span>, <span class="attribute">k</span>=1:nrow(clusgap$Tab))</span><br><span class="line">  p = ggplot(gstab, aes(k, gap)) + geom_line() + geom_point(<span class="attribute">size</span>=5)</span><br><span class="line">  p = p + geom_errorbar(aes(<span class="attribute">ymax</span>=gap+SE.sim, <span class="attribute">ymin</span>=gap-SE.sim))</span><br><span class="line">  p = p + ggtitle(title)</span><br><span class="line">  return(p)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">#Now try out this function. Should work on ordination classes recognized by `scores` function, and provide a ggplot graphic instead of a base graphic. (Special <span class="doctag">Note:</span> the phyloseq-defined `scores` extensions are not exported as regular functions to avoid conflict, so phyloseq-defined `scores` extensions can only be accessed with the `phyloseq:::` namespace prefix in front.)</span></span><br><span class="line">gs = gap_statistic_ordination(exord, <span class="string">"pam1"</span>, <span class="attribute">B</span>=50, <span class="attribute">verbose</span>=<span class="literal">FALSE</span>)</span><br><span class="line"><span class="builtin-name">print</span>(gs, <span class="attribute">method</span>=<span class="string">"Tibs2001SEmax"</span>)</span><br><span class="line"></span><br><span class="line">plot_clusgap(gs)</span><br><span class="line"></span><br><span class="line"><span class="comment">#Base graphics plotting, for comparison.</span></span><br><span class="line">plot(gs, main = <span class="string">"Gap statistic for the 'Enterotypes' data"</span>)</span><br><span class="line">mtext(<span class="string">"Looks like 4 clusters is best, with 3 and 5 close runners up."</span>)</span><br></pre></td></tr></table></figure>

<blockquote>
<p>gap statistic: <a href="https://joey711.github.io/phyloseq/gap-statistic.html" target="_blank" rel="noopener">https://joey711.github.io/phyloseq/gap-statistic.html</a></p>
</blockquote>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/R-Packages/" rel="tag"># R.Packages</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/04/05/2020-11-liaofansixun/" rel="prev" title="2020.11.了凡四训">
      <i class="fa fa-chevron-left"></i> 2020.11.了凡四训
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/04/06/Rpackages-ggcor/" rel="next" title="R包介绍及应用2|ggcor()，主quickcor() + Science相关性组合图">
      R包介绍及应用2|ggcor()，主quickcor() + Science相关性组合图 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  

  </div>


          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#phyloseq简介"><span class="nav-number">1.</span> <span class="nav-text">phyloseq简介</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#教程"><span class="nav-number">2.</span> <span class="nav-text">教程</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Phyloseq包安装"><span class="nav-number">2.1.</span> <span class="nav-text">Phyloseq包安装</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#导入phyloseq数据"><span class="nav-number">2.2.</span> <span class="nav-text">导入phyloseq数据</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#示例（随机生成的数据）："><span class="nav-number">2.2.1.</span> <span class="nav-text">示例（随机生成的数据）：</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#示例（BIOM格式文件）"><span class="nav-number">2.2.2.</span> <span class="nav-text">示例（BIOM格式文件）</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#示例（qiime输出文件）"><span class="nav-number">2.2.3.</span> <span class="nav-text">示例（qiime输出文件）</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#示例（mothur输出文件）"><span class="nav-number">2.2.4.</span> <span class="nav-text">示例（mothur输出文件）</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#pyrotagger文件不作说明，详见教程。"><span class="nav-number">2.2.5.</span> <span class="nav-text">pyrotagger文件不作说明，详见教程。</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#使用phyloseq的例子"><span class="nav-number">2.3.</span> <span class="nav-text">使用phyloseq的例子</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#合并数据"><span class="nav-number">2.4.</span> <span class="nav-text">合并数据</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#示例：merge-samples"><span class="nav-number">2.4.1.</span> <span class="nav-text">示例：merge_samples</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#示例：merge-taxa"><span class="nav-number">2.4.2.</span> <span class="nav-text">示例：merge_taxa</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#示例：merge-phyloseq"><span class="nav-number">2.4.3.</span> <span class="nav-text">示例：merge_phyloseq</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#示例：更复杂的合并"><span class="nav-number">2.4.4.</span> <span class="nav-text">示例：更复杂的合并</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#访问数据和数据预处理"><span class="nav-number">2.5.</span> <span class="nav-text">访问数据和数据预处理</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#Accessors"><span class="nav-number">2.5.1.</span> <span class="nav-text">Accessors</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#preprocessing"><span class="nav-number">2.5.2.</span> <span class="nav-text">preprocessing</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#基于微生物群落数据距离的计算"><span class="nav-number">2.6.</span> <span class="nav-text">基于微生物群落数据距离的计算</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#差异统计"><span class="nav-number">2.7.</span> <span class="nav-text">差异统计</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Yunmeng"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">Yunmeng</p>
  <div class="site-description" itemprop="description">Junior Researcher | VLIZ | Bacteria in Sea Air</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">40</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">15</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/ItsYunmeng" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-github"></i>GitHub</a>
      </span>
  </div>


  <div class="links-of-blogroll motion-element">
    <div class="links-of-blogroll-title">
      <i class="fa fa-fw fa-link"></i>
      Links
    </div>
    <ul class="links-of-blogroll-list">
        <li class="links-of-blogroll-item">
          <a href="http://www.bio-info-trainee.com/" title="http:&#x2F;&#x2F;www.bio-info-trainee.com&#x2F;" rel="noopener" target="_blank">生信菜鸟团</a>
        </li>
    </ul>
  </div>

      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Yunmeng</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> v4.2.0
  </div>
  <span class="post-meta-divider">|</span>
  <div class="theme-info">Theme – <a href="https://pisces.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a> v7.7.2
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
