<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Neo Sans Intel:300,300italic,400,400italic,700,700italic|Times New Roman:300,300italic,400,400italic,700,700italic|Lato:300,300italic,400,400italic,700,700italic|Helvetica:300,300italic,400,400italic,700,700italic|Roboto Mono:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">
<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"itsyunmeng.github.io","root":"/","scheme":"Pisces","version":"7.7.2","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Abstract: Tool for functional abundances prediction of marker gene sequences.s">
<meta property="og:type" content="article">
<meta property="og:title" content="16S_PICRUSt2 - predicting functional abundances based on marker gene sequences">
<meta property="og:url" content="https://itsyunmeng.github.io/2020/07/15/16S-PICRUSt2/index.html">
<meta property="og:site_name" content="Yunmeng&#39;s Blog">
<meta property="og:description" content="Abstract: Tool for functional abundances prediction of marker gene sequences.s">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-07-15T08:16:48.000Z">
<meta property="article:modified_time" content="2020-08-12T14:14:11.864Z">
<meta property="article:author" content="Yunmeng">
<meta property="article:tag" content="Linux">
<meta property="article:tag" content="16S">
<meta property="article:tag" content="PICRUSt2">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://itsyunmeng.github.io/2020/07/15/16S-PICRUSt2/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true
  };
</script>

  <title>16S_PICRUSt2 - predicting functional abundances based on marker gene sequences | Yunmeng's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <div>
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Yunmeng's Blog</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>


<nav class="site-nav">
  
  <ul id="menu" class="menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-fw fa-home"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-fw fa-user"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-fw fa-tags"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-fw fa-th"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-fw fa-archive"></i>Archives</a>

  </li>
  </ul>

</nav>
</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content">
            

  <div class="posts-expand">
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block " lang="en">
    <link itemprop="mainEntityOfPage" href="https://itsyunmeng.github.io/2020/07/15/16S-PICRUSt2/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="Yunmeng">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Yunmeng's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          16S_PICRUSt2 - predicting functional abundances based on marker gene sequences
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-07-15 10:16:48" itemprop="dateCreated datePublished" datetime="2020-07-15T10:16:48+02:00">2020-07-15</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-calendar-check-o"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2020-08-12 16:14:11" itemprop="dateModified" datetime="2020-08-12T16:14:11+02:00">2020-08-12</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Omics/" itemprop="url" rel="index"><span itemprop="name">Omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Omics/Metabarcoding/" itemprop="url" rel="index"><span itemprop="name">Metabarcoding</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>Abstract:</strong> Tool for functional abundances prediction of marker gene sequences.s</p>
<a id="more"></a>
<h1>Introduction</h1>
<p>PICRUSt2 (Phylogenetic Investigation of Communities by Reconstruction of Unobserved States) is a software for predicting functional abundances based only on marker gene sequences.<br>
<u>“Function” usually refers to gene families such as KEGG orthologs and Enzyme Classification numbers, but predictions can be made for any arbitrary trait.</u> Similarly, <u>predictions are typically based on 16S rRNA gene sequencing data</u>, but other marker genes can also be used.<br>
PICRUSt2 includes these and other improvements over the original version:</p>
<ul>
<li>Allow users to predict functions for any <strong>16S</strong> sequences. <u>Representative sequences from OTUs or amplicon sequence variants (e.g. DADA2 and deblur output)</u> can be used as input by taking a sequence placement approach</li>
<li>Database of reference genomes used for prediction has been expanded by &gt;10X.</li>
<li>Addition of hidden-state prediction algorithms from the castor R package.</li>
<li>Allows output of MetaCyc ontology predictions that will be comparable with common shotgun metagenomics outputs.</li>
<li>Inference of pathway abundances now relies on MinPath, which makes these predictions more stringent.</li>
</ul>
<p><strong>Methods:</strong> sequence placement, hidden state prediction, metagenome prediction, infer pathway abundances.<br>
<strong>input files:</strong> ASV sequence file and abundance table.<br>
<strong>output files:</strong> gene family and pathway abundances.</p>
<h1>Installation</h1>
<figure class="highlight angelscript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">conda create -n picrust2 -c bioconda -c conda-forge picrust2=<span class="number">2.3</span><span class="number">.0</span>_b</span><br><span class="line">conda activate picrust2</span><br></pre></td></tr></table></figure>
<h1>Data</h1>
<p>total absolute abundance table: <code>feature-table.biom</code><br>
represent sequences: <code>dna-sequences.fasta</code></p>
<h1>Run</h1>
<h2 id="Sequence-placement">Sequence placement</h2>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">place_seqs<span class="selector-class">.py</span> -s results/dna-sequences<span class="selector-class">.fasta</span> -o placed_seqs<span class="selector-class">.tre</span> -<span class="selector-tag">p</span> <span class="number">13</span> --intermediate placement_working</span><br></pre></td></tr></table></figure>
<p><strong>output files</strong></p>
<ul>
<li><code>placed_seqs.tre</code> - Tree of reference and study 16S sequences</li>
</ul>
<h2 id="Hidden-state-prediction">Hidden state prediction</h2>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">hsp<span class="selector-class">.py</span> -<span class="selector-tag">i</span> <span class="number">16</span>S -t placed_seqs<span class="selector-class">.tre</span> -o marker_nsti_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -<span class="selector-tag">p</span> <span class="number">13</span> -n</span><br><span class="line">hsp<span class="selector-class">.py</span> -<span class="selector-tag">i</span> EC -t placed_seqs<span class="selector-class">.tre</span> -o EC_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -<span class="selector-tag">p</span> <span class="number">13</span></span><br><span class="line">hsp<span class="selector-class">.py</span> -<span class="selector-tag">i</span> KO -t placed_seqs<span class="selector-class">.tre</span> -o KO_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -<span class="selector-tag">p</span> <span class="number">13</span></span><br></pre></td></tr></table></figure>
<p><strong>output files</strong></p>
<ul>
<li><code>marker_nsti_predicted.tsv.gz</code> - Predicted 16S copy numbers and NSTI per ASV.</li>
<li><code>EC_predicted.tsv.gz</code> - Predicted EC number copy numbers per ASV.</li>
<li><code>KO_predicted.tsv.gz</code> - As EC_predicted.tsv.gz above, but for KO predictions.</li>
</ul>
<h2 id="Metagenome-prediction">Metagenome prediction</h2>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">metagenome_pipeline<span class="selector-class">.py</span> -<span class="selector-tag">i</span> results/feature_table/feature-<span class="selector-tag">table</span><span class="selector-class">.biom</span> \</span><br><span class="line">                       -m marker_nsti_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> \</span><br><span class="line">                       -f EC_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> \</span><br><span class="line">                       -o EC_metagenome_out</span><br><span class="line">metagenome_pipeline<span class="selector-class">.py</span> -<span class="selector-tag">i</span> results/feature_table/feature-<span class="selector-tag">table</span><span class="selector-class">.biom</span> \</span><br><span class="line">                       -m marker_nsti_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> \</span><br><span class="line">                       -f KO_predicted<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> \</span><br><span class="line">                       -o KO_metagenome_out</span><br></pre></td></tr></table></figure>
<p><strong>output files</strong></p>
<ul>
<li><code>EC_metagenome_out</code> - Folder containing unstratified EC number metagenome predictions (pred_metagenome_unstrat.tsv.gz), sequence table normalized by predicted 16S copy number abundances (seqtab_norm.tsv.gz), and the per-sample NSTI values weighted by the abundance of each ASV (weighted_nsti.tsv.gz).</li>
<li><code>KO_metagenome_out</code> - As EC_metagenome_out above, but for KO metagenomes.</li>
</ul>
<p>#Infer pathway abundances</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">pathway_pipeline<span class="selector-class">.py</span> -<span class="selector-tag">i</span> EC_metagenome_out/pred_metagenome_unstrat<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -o pathways_out --intermediate minpath_working -<span class="selector-tag">p</span> <span class="number">13</span></span><br></pre></td></tr></table></figure>
<p><strong>output files</strong></p>
<ul>
<li><code>pathways_out</code> - <u>Folder containing predicted pathway abundances and coverages per-sample, based on predicted EC number abundances.</u></li>
</ul>
<h2 id="Add-descriptions">Add descriptions</h2>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">add_descriptions<span class="selector-class">.py</span> -<span class="selector-tag">i</span> EC_metagenome_out/pred_metagenome_unstrat<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -m ec -o EC_metagenome_out/pred_metagenome_unstrat_descrip<span class="selector-class">.tsv</span>.gz</span><br><span class="line">add_descriptions<span class="selector-class">.py</span> -<span class="selector-tag">i</span> KO_metagenome_out/pred_metagenome_unstrat<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -m KO -o KO_metagenome_out/pred_metagenome_unstrat_descrip<span class="selector-class">.tsv</span>.gz</span><br><span class="line">add_descriptions<span class="selector-class">.py</span> -<span class="selector-tag">i</span> pathways_out/path_abun_unstrat<span class="selector-class">.tsv</span><span class="selector-class">.gz</span> -m METACYC -o pathways_out/path_abun_unstrat_descrip<span class="selector-class">.tsv</span>.gz</span><br></pre></td></tr></table></figure>
<p>Ref:</p>
<blockquote>
<p>picrust2: <a href="https://github.com/picrust/picrust2/wiki" target="_blank" rel="noopener">https://github.com/picrust/picrust2/wiki</a></p>
</blockquote>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/Linux/" rel="tag"># Linux</a>
              <a href="/tags/16S/" rel="tag"># 16S</a>
              <a href="/tags/PICRUSt2/" rel="tag"># PICRUSt2</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/07/14/biosoft-STAMP/" rel="prev" title="STAMP - a software package for analyzing metagenomic profiles that promotes ‘best practices’ in choosing appropriate statistical techniques and reporting results">
      <i class="fa fa-chevron-left"></i> STAMP - a software package for analyzing metagenomic profiles that promotes ‘best practices’ in choosing appropriate statistical techniques and reporting results
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/08/05/tutorial-LaTeX/" rel="next" title="LaTeX基础入门">
      LaTeX基础入门 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  

  </div>


          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">1.</span> <span class="nav-text">Introduction</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">2.</span> <span class="nav-text">Installation</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">3.</span> <span class="nav-text">Data</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">4.</span> <span class="nav-text">Run</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Sequence-placement"><span class="nav-number">4.1.</span> <span class="nav-text">Sequence placement</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Hidden-state-prediction"><span class="nav-number">4.2.</span> <span class="nav-text">Hidden state prediction</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Metagenome-prediction"><span class="nav-number">4.3.</span> <span class="nav-text">Metagenome prediction</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Add-descriptions"><span class="nav-number">4.4.</span> <span class="nav-text">Add descriptions</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Yunmeng"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">Yunmeng</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">71</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">21</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">21</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/ItsYunmeng" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-github"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://twitter.com/ItsYunmeng" title="Twitter → https:&#x2F;&#x2F;twitter.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-twitter"></i>Twitter</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://www.instagram.com/itsyunmeng" title="Instagram → https:&#x2F;&#x2F;www.instagram.com&#x2F;itsyunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-instagram"></i>Instagram</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yunmengli@hotmail.com" title="E-Mail → mailto:yunmengli@hotmail.com" rel="noopener" target="_blank"><i class="fa fa-fw fa-envelope"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Yunmeng</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> v4.2.0
  </div>
  <span class="post-meta-divider">|</span>
  <div class="theme-info">Theme – <a href="https://pisces.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a> v7.7.2
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  













<script>
if (document.querySelectorAll('pre.mermaid').length) {
  NexT.utils.getScript('//cdn.jsdelivr.net/npm/mermaid@8/dist/mermaid.min.js', () => {
    mermaid.initialize({
      theme    : 'forest',
      logLevel : 3,
      flowchart: { curve     : 'linear' },
      gantt    : { axisFormat: '%m/%d/%Y' },
      sequence : { actorMargin: 50 }
    });
  }, window.mermaid);
}
</script>


  

  

  

</body>
</html>
