<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Neo Sans Intel:300,300italic,400,400italic,700,700italic|Times New Roman:300,300italic,400,400italic,700,700italic|Lato:300,300italic,400,400italic,700,700italic|Helvetica:300,300italic,400,400italic,700,700italic|Roboto Mono:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">
<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"itsyunmeng.github.io","root":"/","scheme":"Gemini","version":"7.7.2","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":true},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="Abstract: This documents shows the use of qiime2 to process metabarcoding data.">
<meta property="og:type" content="article">
<meta property="og:title" content="Using QIIME2 to process 16S rRNA metabarcoding illumina Miseq sequencing data">
<meta property="og:url" content="https://itsyunmeng.github.io/2020/07/03/bioplatform-qiime2/index.html">
<meta property="og:site_name" content="Yunmeng&#39;s Blog">
<meta property="og:description" content="Abstract: This documents shows the use of qiime2 to process metabarcoding data.">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-07-02T22:33:38.000Z">
<meta property="article:modified_time" content="2020-08-12T17:03:50.138Z">
<meta property="article:author" content="Yunmeng">
<meta property="article:tag" content="qiime2">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://itsyunmeng.github.io/2020/07/03/bioplatform-qiime2/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true
  };
</script>

  <title>Using QIIME2 to process 16S rRNA metabarcoding illumina Miseq sequencing data | Yunmeng's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <div>
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Yunmeng's Blog</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>


<nav class="site-nav">
  
  <ul id="menu" class="menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-fw fa-home"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-fw fa-user"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-fw fa-tags"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-fw fa-th"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-fw fa-archive"></i>Archives</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>

</nav>
  <div class="site-search">
    <div class="search-pop-overlay">
  <div class="popup search-popup">
      <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocorrect="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

  </div>
</div>

  </div>
</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content">
            

  <div class="posts-expand">
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block " lang="en">
    <link itemprop="mainEntityOfPage" href="https://itsyunmeng.github.io/2020/07/03/bioplatform-qiime2/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="Yunmeng">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Yunmeng's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Using QIIME2 to process 16S rRNA metabarcoding illumina Miseq sequencing data
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-07-03 00:33:38" itemprop="dateCreated datePublished" datetime="2020-07-03T00:33:38+02:00">2020-07-03</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-calendar-check-o"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2020-08-12 19:03:50" itemprop="dateModified" datetime="2020-08-12T19:03:50+02:00">2020-08-12</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Omics/" itemprop="url" rel="index"><span itemprop="name">Omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Omics/Metabarcoding/" itemprop="url" rel="index"><span itemprop="name">Metabarcoding</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>Abstract:</strong> This documents shows the use of qiime2 to process metabarcoding data.</p>
<a id="more"></a>
<h1>Environment</h1>
<ul>
<li>Ubuntu 16.04.6 LTS #cat /etc/issue</li>
<li>qiime2-2020.2</li>
</ul>
<blockquote>
<p>qiime2 install: <a href="https://docs.qiime2.org/2020.2/install/native/" target="_blank" rel="noopener">https://docs.qiime2.org/2020.2/install/native/</a></p>
</blockquote>
<h1>Data</h1>
<ul>
<li><code>raw_data</code> folder: 1) paired-end illumina Miseq sequencing data of 16S rRNA gene in the format of <code>*.fastq</code>; 2) <code>manifest.txt</code>; 3) <code>mapping.txt</code>.</li>
<li>classifier: <code>gg-13-8-99-nb-classifier.qza</code></li>
</ul>
<h1>Workflow</h1>
<h2 id="Check-the-quality-of-raw-reads">Check the quality of raw reads</h2>
<figure class="highlight properties"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="attr">mkdir</span> <span class="string">fastqc_out</span></span><br><span class="line"><span class="attr">fastqc</span> <span class="string">raw_data/*.fastq -o fasqc_out/</span></span><br><span class="line"><span class="attr">cd</span> <span class="string">fasqc_out/</span></span><br><span class="line"><span class="attr">multiqc</span> <span class="string">.</span></span><br></pre></td></tr></table></figure>
<p>-&gt; get multiqc_report.html -&gt; see if most of <u>per sequence quality scores</u> is above 30</p>
<blockquote>
<p>multiqc: <a href="https://github.com/ewels/MultiQC" target="_blank" rel="noopener">https://github.com/ewels/MultiQC</a></p>
</blockquote>
<h2 id="Importing-data">Importing data</h2>
<figure class="highlight sas"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mkdir results</span><br><span class="line">qiime tools import --type <span class="string">'SampleData[PairedEndSequencesWithQuality]'</span> --<span class="meta">input</span>-path raw_data/manifest.txt --<span class="meta">output</span>-path results/paired-<span class="meta">end</span>-demux.qza --<span class="meta">input</span>-<span class="meta">format</span> PairedEndFastqManifestPhred33</span><br></pre></td></tr></table></figure>
<p>-&gt; get paired-end-demux.qza file</p>
<h2 id="Trim-adapter-optional">Trim adapter (optional)</h2>
<p><code>--p-discard-untrimmed</code>: discard reads in which no adapter was found.<br>
<code>--p-no-indels</code>: don’t allow insertions or deletions of bases when matching adapters</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">cutadapt</span> <span class="comment">trim</span><span class="literal">-</span><span class="comment">paired</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">demultiplexed</span><span class="literal">-</span><span class="comment">sequences</span> <span class="comment">paired</span><span class="literal">-</span><span class="comment">end</span><span class="literal">-</span><span class="comment">demux</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">cores</span> <span class="comment">8</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">front</span><span class="literal">-</span><span class="comment">f</span> <span class="comment">CCTACGGGNGGCWGCAG</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">front</span><span class="literal">-</span><span class="comment">r</span> <span class="comment">GACTACHVGGGTATCTAATCC</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">discard</span><span class="literal">-</span><span class="comment">untrimmed</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">no</span><span class="literal">-</span><span class="comment">indels</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">trimmed</span><span class="literal">-</span><span class="comment">sequences</span> <span class="comment">reads_trimmed</span><span class="string">.</span><span class="comment">qza</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">demux</span> <span class="comment">summarize</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">data</span> <span class="comment">reads_trimmed</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">reads_trimmed</span><span class="string">.</span><span class="comment">qzv</span></span><br></pre></td></tr></table></figure>
<p><code>*.fastq*</code> here don’t contain primer, so we don’t do this procedure.</p>
<h2 id="Quality-control">Quality control</h2>
<h3 id="stats-demux">stats_demux</h3>
<figure class="highlight armasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="symbol">qiime</span> demux summarize --i-<span class="meta">data</span> results/paired-<span class="meta">end</span>-demux.qza --o-visualization results/paired-<span class="meta">end</span>-demux.qzv</span><br></pre></td></tr></table></figure>
<p>-&gt; get paired-end-demux.qzv -&gt; visualization</p>
<blockquote>
<p>Importing data: <a href="https://docs.qiime2.org/2020.6/tutorials/importing/" target="_blank" rel="noopener">https://docs.qiime2.org/2020.6/tutorials/importing/</a><br>
Visualization: <a href="https://view.qiime2.org/" target="_blank" rel="noopener">https://view.qiime2.org/</a></p>
</blockquote>
<h3 id="stats-dada2">stats_dada2</h3>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">dada2</span> <span class="comment">denoise</span><span class="literal">-</span><span class="comment">paired</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">demultiplexed</span><span class="literal">-</span><span class="comment">seqs</span> <span class="comment">results/paired</span><span class="literal">-</span><span class="comment">end</span><span class="literal">-</span><span class="comment">demux</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">trunc</span><span class="literal">-</span><span class="comment">len</span><span class="literal">-</span><span class="comment">f</span> <span class="comment">220</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">trunc</span><span class="literal">-</span><span class="comment">len</span><span class="literal">-</span><span class="comment">r</span> <span class="comment">230</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">trim</span><span class="literal">-</span><span class="comment">left</span><span class="literal">-</span><span class="comment">f</span> <span class="comment">0</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">trim</span><span class="literal">-</span><span class="comment">left</span><span class="literal">-</span><span class="comment">r</span> <span class="comment">0</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">trunc</span><span class="literal">-</span><span class="comment">q</span> <span class="comment">5</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">chimera</span><span class="literal">-</span><span class="comment">method</span> <span class="comment">pooled</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">table</span> <span class="comment">results/table</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">representative</span><span class="literal">-</span><span class="comment">sequences</span> <span class="comment">results/rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">denoising</span><span class="literal">-</span><span class="comment">stats</span> <span class="comment">results/denoising</span><span class="literal">-</span><span class="comment">stats</span><span class="string">.</span><span class="comment">qza</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">metadata</span> <span class="comment">tabulate</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">input</span><span class="literal">-</span><span class="comment">file</span> <span class="comment">results/denoising</span><span class="literal">-</span><span class="comment">stats</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/denoising</span><span class="literal">-</span><span class="comment">stats</span><span class="string">.</span><span class="comment">qzv</span></span><br></pre></td></tr></table></figure>
<p>-&gt; get denoising-stats.qza -&gt; get <code>*.qzv</code> -&gt; visualization</p>
<p>in addition,<br>
-&gt; get table.qza =&gt; see ##abundance table<br>
-&gt; get rep-seqs.qza =&gt; see ##sequence alignment and taxonomic assignment</p>
<h2 id="Filter-ASV-optional">Filter ASV (optional)</h2>
<h3 id="don’t-filter">don’t filter</h3>
<blockquote>
<p>“I don’t consider filtering features to be an essential step, especially given how well the modern denoisers (such as DADA2 and Deblur) work.</p>
</blockquote>
<h3 id="filter">filter</h3>
<blockquote>
<p>“When I do filter features,  (a) by requiring them to show up in some minimum number of samples, or (b) if they don’t achieve some minimum level of taxonomy assignment.”</p>
</blockquote>
<p>详见参考中的 <em>扩增子分析流程</em></p>
<ul>
<li><code>qiime feature-table filter-features</code></li>
<li><code>qiime taxa filter-table</code></li>
</ul>
<h2 id="Sequence-alignment">Sequence alignment</h2>
<p>-&gt; get rep-seqs.qza -&gt; get <code>*.qzv</code> -&gt; visualization</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">feature</span><span class="literal">-</span><span class="comment">table</span> <span class="comment">tabulate</span><span class="literal">-</span><span class="comment">seqs</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">data</span> <span class="comment">results/rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qzv</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">tools</span> <span class="comment">export</span> --<span class="comment">input</span><span class="literal">-</span><span class="comment">path</span> <span class="comment">results/rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">output</span><span class="literal">-</span><span class="comment">path</span> <span class="comment">results/</span></span><br></pre></td></tr></table></figure>
<p>=&gt; <code>rep-seqs.qzv</code>中是每个ASV的序列，点击序列可以直接跳转到blast，非常方便。<br>
=&gt; get <code>dna-sequences.fasta</code> file</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">alignment</span> <span class="comment">mafft</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">sequences</span> <span class="comment">results/rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">alignment</span> <span class="comment">results/aligned</span><span class="literal">-</span><span class="comment">rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">alignment</span> <span class="comment">mask</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">alignment</span> <span class="comment">results/aligned</span><span class="literal">-</span><span class="comment">rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">masked</span><span class="literal">-</span><span class="comment">alignment</span> <span class="comment">results/masked</span><span class="literal">-</span><span class="comment">aligned</span><span class="literal">-</span><span class="comment">rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">phylogeny</span> <span class="comment">fasttree</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">alignment</span> <span class="comment">results/masked</span><span class="literal">-</span><span class="comment">aligned</span><span class="literal">-</span><span class="comment">rep</span><span class="literal">-</span><span class="comment">seqs</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">tree</span> <span class="comment">results/unrooted</span><span class="literal">-</span><span class="comment">tree</span><span class="string">.</span><span class="comment">qza</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">phylogeny</span> <span class="comment">midpoint</span><span class="literal">-</span><span class="comment">root</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">tree</span> <span class="comment">results/unrooted</span><span class="literal">-</span><span class="comment">tree</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">rooted</span><span class="literal">-</span><span class="comment">tree</span> <span class="comment">results/rooted</span><span class="literal">-</span><span class="comment">tree</span><span class="string">.</span><span class="comment">qza</span></span><br></pre></td></tr></table></figure>
<p>-&gt; get <code>unrooted-tree.qza</code> and <code>rooted-tree.qza</code></p>
<figure class="highlight dos"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">qiime tools export --input-<span class="built_in">path</span> results/rooted-<span class="built_in">tree</span>.qza --output-<span class="built_in">path</span> results/rooted-<span class="built_in">tree</span></span><br><span class="line">qiime tools export --input-<span class="built_in">path</span> results/unrooted-<span class="built_in">tree</span>.qza --output-<span class="built_in">path</span> results/unrooted-<span class="built_in">tree</span></span><br></pre></td></tr></table></figure>
<p>-&gt; get <code>unrooted-tree.nwk</code> and <code>rooted-tree.nwk</code></p>
<h2 id="Taxonomic-assignment">Taxonomic assignment</h2>
<figure class="highlight angelscript"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime feature-classifier classify-sklearn --i-classifier classifiers/gg<span class="number">-13</span><span class="number">-8</span><span class="number">-99</span>-nb-classifier.qza --i-reads results/rep-seqs.qza --o-classification results/taxonomy.qza</span><br></pre></td></tr></table></figure>
<blockquote>
<p>if the version of scikit-learn is not match, <code>pip install -U scikit-learn==version No.</code><br>
Installing scikit-learn: <a href="http://ogrisel.github.io/scikit-learn.org/sklearn-tutorial/install.html" target="_blank" rel="noopener">http://ogrisel.github.io/scikit-learn.org/sklearn-tutorial/install.html</a></p>
</blockquote>
<p>-&gt; get taxonomy.qza -&gt; get <code>*.qzv</code> -&gt; visualization</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">tools</span> <span class="comment">export</span> --<span class="comment">input</span><span class="literal">-</span><span class="comment">path</span> <span class="comment">results/taxonomy</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">output</span><span class="literal">-</span><span class="comment">path</span> <span class="comment">results/taxonomy</span></span><br><span class="line"><span class="comment">qiime</span> <span class="comment">metadata</span> <span class="comment">tabulate</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">input</span><span class="literal">-</span><span class="comment">file</span> <span class="comment">results/taxonomy</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/taxonomy</span><span class="string">.</span><span class="comment">qzv</span></span><br></pre></td></tr></table></figure>
<p>-&gt; get <code>taxonomy.tsv</code> and <code>taxonomy.qzv</code></p>
<h2 id="Phylogenetic-tree">Phylogenetic tree</h2>
<p>=&gt; <code>rooted-tree.qza</code> and <code>rooted-tree.nwk</code> both can be imported into iTOL to build phylogenetic tree. using <code>taxonomy.qza</code> as annotation file.</p>
<blockquote>
<p>iTOL: <a href="https://itol.embl.de/tree/21222422923956981594160007" target="_blank" rel="noopener">https://itol.embl.de/tree/21222422923956981594160007</a></p>
</blockquote>
<h2 id="Diversity">Diversity</h2>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime feature-<span class="keyword">table</span> <span class="keyword">summarize</span> --i-<span class="keyword">table</span> results/<span class="keyword">table</span>.qza --<span class="keyword">m</span>-<span class="keyword">sample</span>-metadata-<span class="keyword">file</span> raw_data/mapping.txt --o-visualization results/<span class="keyword">table</span>.qzv</span><br></pre></td></tr></table></figure>
<p>=&gt; 注意生成的这个table.qzv (100%相似程度)，需要通过它来了解，在测序深度选择多少合适。</p>
<p>-&gt; get core-metrics</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mkdir results/<span class="number">2</span>.Diversity</span><br><span class="line">qiime diversity core-metrics-phylogenetic --i-phylogeny results/rooted-tree<span class="selector-class">.qza</span> --i-<span class="selector-tag">table</span> results/<span class="selector-tag">table</span><span class="selector-class">.qza</span> --p-sampling-depth <span class="number">1132</span> --m-metadata-file raw_data/mapping<span class="selector-class">.txt</span> --output-dir results/core-metrics</span><br></pre></td></tr></table></figure>
<h3 id="Alpha-diversity">Alpha diversity</h3>
<h4 id="alpha-rarefaction">alpha rarefaction</h4>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">diversity</span> <span class="comment">alpha</span><span class="literal">-</span><span class="comment">rarefaction</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">table</span> <span class="comment">results/table</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">max</span><span class="literal">-</span><span class="comment">depth</span> <span class="comment">30000</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">phylogeny</span> <span class="comment">results/rooted</span><span class="literal">-</span><span class="comment">tree</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">metadata</span><span class="literal">-</span><span class="comment">file</span> <span class="comment">raw_data/mapping</span><span class="string">.</span><span class="comment">txt</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/rare</span><span class="string">.</span><span class="comment">qzv</span></span><br></pre></td></tr></table></figure>
<h4 id="significance-analysis-kruskal-Wallis-test">significance analysis (kruskal-Wallis test)</h4>
<p>-&gt; visualization</p>
<figure class="highlight gradle"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime diversity alpha-<span class="keyword">group</span>-significance --i-alpha-diversity results<span class="regexp">/core-metrics/</span>observed_otus_vector.qza --m-metadata-<span class="keyword">file</span> raw_data<span class="regexp">/mapping.txt --o-visualization results/</span>core-metrics<span class="regexp">/observed_otus_vector.qzv</span></span><br></pre></td></tr></table></figure>
<h3 id="Beta-diversity">Beta diversity</h3>
<h4 id="PCoA-plot">PCoA plot</h4>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">emperor</span> <span class="comment">plot</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">pcoa</span> <span class="comment">results/co</span><span class="literal">-</span><span class="comment">metrics/weighted_unifrac_pcoa_results</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">metadata</span><span class="literal">-</span><span class="comment">file</span> <span class="comment">raw_data/mapping</span><span class="string">.</span><span class="comment">txt</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/co</span><span class="literal">-</span><span class="comment">metrics/weighted_unifrac_emperor</span><span class="string">.</span><span class="comment">qzv</span></span><br></pre></td></tr></table></figure>
<h4 id="significance-analysis-PERMANOVA-and-paired-comparison">significance analysis (PERMANOVA and paired comparison)</h4>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">diversity</span> <span class="comment">beta</span><span class="literal">-</span><span class="comment">group</span><span class="literal">-</span><span class="comment">significance</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">distance</span><span class="literal">-</span><span class="comment">matrix</span> <span class="comment">results/core</span><span class="literal">-</span><span class="comment">metrics/unweighted_unifrac_distance_matrix</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">metadata</span><span class="literal">-</span><span class="comment">file</span> <span class="comment">raw_data/mapping</span><span class="string">.</span><span class="comment">txt</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">metadata</span><span class="literal">-</span><span class="comment">column</span> <span class="comment">group2</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/core</span><span class="literal">-</span><span class="comment">metrics/unweighted_unifrac_significance</span><span class="string">.</span><span class="comment">qzv</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">pairwise</span></span><br></pre></td></tr></table></figure>
<h2 id="Abundance-table">Abundance table</h2>
<h3 id="Absolute-abundance">Absolute abundance</h3>
<p>-&gt; get total absolute abundance table with taxonomy information</p>
<figure class="highlight gams"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">qiime tools export --input-path results/<span class="keyword">table</span>.qza <span class="comment">--output-path feature-table</span></span><br><span class="line">biom convert -i results/feature_table/feature-<span class="keyword">table</span>.biom <span class="comment">-o results</span>/feature_table/<span class="comment">feature-table.txt --to-tsv</span></span><br><span class="line">biom add-metadata -i results/feature_table/feature-<span class="keyword">table</span>.biom <span class="comment">-o results</span>/feature_table/<span class="comment">feature-table-tax.biom --observation-metadata-fp results</span>/taxonomy/<span class="comment">taxonomy.tsv --observation-header OTUID,taxonomy,confidence</span></span><br><span class="line">biom convert -i results/feature-<span class="keyword">table</span>/feature-<span class="keyword">table</span>-tax.biom <span class="comment">-o results</span>/feature-<span class="keyword">table</span>/<span class="comment">feature-table-tax.txt --to-tsv --header-key taxonomy</span></span><br></pre></td></tr></table></figure>
<p>-&gt; get absolute abundance table with taxonomy information at different level (e.g. Phylum)</p>
<figure class="highlight gams"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">qiime taxa collapse --i-<span class="keyword">table</span> results/<span class="keyword">table</span>.qza <span class="comment">--i-taxonomy results</span>/taxonomy.qza --p-level <span class="number">2</span> --o-collapsed-<span class="keyword">table</span> results/<span class="comment">table-phylum.qza</span></span><br><span class="line">qiime tools export --input-path results/<span class="keyword">table</span>-phylum.qza <span class="comment">--output-path results</span>/feature_table/<span class="comment">phylum</span></span><br><span class="line">biom convert -i results/feature_table/phylum/feature-<span class="keyword">table</span>.biom <span class="comment">-o results</span>/feature_table/<span class="comment">phylum</span>/feature-<span class="keyword">table</span>.txt --to-tsv</span><br></pre></td></tr></table></figure>
<p>-&gt; get absolute abundance table with taxonomy information at different level (e.g. phylum) and treatment (e.g. group2)</p>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">qiime feature-<span class="keyword">table</span> group --i-<span class="keyword">table</span> results/<span class="keyword">table</span>-phylum.qza --p-axis <span class="keyword">sample</span> --p-mode <span class="keyword">sum</span> --<span class="keyword">m</span>-metadata-<span class="keyword">file</span> raw_data/mapping.txt --<span class="keyword">m</span>-metadata-column group2 --o-grouped-<span class="keyword">table</span> results/<span class="keyword">table</span>-phylum-group2.qza</span><br><span class="line">qiime metadata <span class="keyword">tabulate</span> --<span class="keyword">m</span>-<span class="keyword">input</span>-<span class="keyword">file</span> results/<span class="keyword">table</span>-phylum-group2.qza --o-visualization results/<span class="keyword">table</span>-phylum-group2.qzv</span><br><span class="line">qiime tools export --<span class="keyword">input</span>-path results/<span class="keyword">table</span>-phylum-group2.qza --output-path results/<span class="keyword">table</span>-phylum-group2</span><br><span class="line">biom convert -i results/<span class="keyword">table</span>-phylum-group2/feature-<span class="keyword">table</span>.biom -o results/<span class="keyword">table</span>-phylum-group2/feature-<span class="keyword">table</span>-group2.txt --to-tsv</span><br></pre></td></tr></table></figure>
<h3 id="Relative-abundance">Relative abundance</h3>
<p>-&gt; get total relative abundance table with taxonomy information</p>
<figure class="highlight gams"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">qiime feature-<span class="keyword">table</span> relative-frequency <span class="comment">--i-table results</span>/<span class="keyword">table</span>.qza --o-relative-frequency-<span class="keyword">table</span> results/<span class="comment">table-relative.qza</span></span><br><span class="line">qiime tools export --input-path results/<span class="keyword">table</span>-relative.qza <span class="comment">--output-path results</span>/feature_table_relative</span><br><span class="line">biom convert -i results/<span class="comment">feature_table_relative</span>/feature-<span class="keyword">table</span>.biom -o results/<span class="comment">feature_table_relative</span>/feature-<span class="keyword">table</span>-relative.txt --to-tsv</span><br><span class="line">biom add-metadata -i results/<span class="comment">feature_table_relative</span>/feature-<span class="keyword">table</span>.biom -o results/<span class="comment">feature_table_relative</span>/feature-<span class="keyword">table</span>-relative-taxonomy.biom --observation-metadata-fp results/<span class="comment">taxonomy</span>/taxonomy.tsv --observation-header OTUID,taxonomy,confidence</span><br><span class="line">biom convert -i results/<span class="comment">feature_table_relative</span>/feature-<span class="keyword">table</span>-relative-taxonomy.biom -o results/<span class="comment">feature_table_relative</span>/feature-<span class="keyword">table</span>-relative-taxonomy.txt --to-tsv --header-key taxonomy</span><br></pre></td></tr></table></figure>
<p>-&gt; get relative abundance table with taxonomy information at different level (e.g. Phylum)<br>
–&gt; statistic absolute abundance table to get the relative abundance table at different level<br>
–&gt; see below</p>
<figure class="highlight gams"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">qiime feature-<span class="keyword">table</span> relative-frequency <span class="comment">--i-table results</span>/<span class="keyword">table</span>-phylum.qza --o-relative-frequency-<span class="keyword">table</span> results/<span class="comment">feature_table_relative</span>/phylum/<span class="comment">table-relative-phylum.qza</span></span><br><span class="line">qiime tools export --input-path results/feature_table_relative/phylum/<span class="keyword">table</span>-relative-phylum.qza <span class="comment">--output-path results</span>/feature_table_relative/<span class="comment">phylum</span>/</span><br><span class="line">biom convert -i results/<span class="comment">feature_table_relative</span>/phylum/<span class="comment">feature-table.biom -o results</span>/feature_table_relative/<span class="comment">phylum</span>/feature-<span class="keyword">table</span>-phylum.txt --to-tsv</span><br></pre></td></tr></table></figure>
<h3 id="Relative-abundance-barplot">Relative abundance barplot</h3>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime taxa barplot --i-<span class="selector-tag">table</span> results/<span class="selector-tag">table</span><span class="selector-class">.qza</span> --i-taxonomy results/taxonomy<span class="selector-class">.qza</span> --m-metadata-file raw_data/mapping<span class="selector-class">.txt</span> --o-visualization results/taxa_barplot.qzv</span><br></pre></td></tr></table></figure>
<h3 id="Absolute-abundance-heatmap">Absolute abundance heatmap</h3>
<p>phylum level &amp; treatment group2</p>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime feature-<span class="keyword">table</span> heatmap --i-<span class="keyword">table</span> results/<span class="keyword">table</span>-phylum.qza --output-<span class="keyword">dir</span> results/phylum-heatmap --<span class="keyword">m</span>-<span class="keyword">sample</span>-metadata-<span class="keyword">file</span> raw_data/mapping.txt --<span class="keyword">m</span>-<span class="keyword">sample</span>-metadata-column group2</span><br></pre></td></tr></table></figure>
<h3 id="Abundance-comparison-ANCOM">Abundance comparison (ANCOM)</h3>
<p>-&gt; different treatment (e.g. group2)</p>
<figure class="highlight smali"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">qiime composition<span class="built_in"> add-pseudocount </span>--i-table results/table.qza --o-composition-table results/comp-table.qza</span><br><span class="line">qiime composition ancom --i-table results/comp-table.qza --m-metadata-file raw_data/mapping.txt --m-metadata-column group2 --o-visualization results/ancom-group2.qzv</span><br></pre></td></tr></table></figure>
<p>-&gt; different taxonomy level (e.g. Phylum)</p>
<figure class="highlight smali"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">qiime composition<span class="built_in"> add-pseudocount </span>--i-table results/table-12.qza --o-composition-table results/comp-table-phylum.qza</span><br><span class="line">qiime composition ancom --i-table results/comp-table-phylum.qza --m-metadata-file raw_data/mapping.txt --m-metadata-column group2 --o-visualization results/phylum-ancom-group2.qzv</span><br></pre></td></tr></table></figure>
<h3 id="Differential-abundance-analysis">Differential abundance analysis</h3>
<p>-&gt; which factor result in the variety of relative abundance?<br>
correlation-clustering</p>
<figure class="highlight matlab"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">qiime gneiss correlation-clustering --<span class="built_in">i</span>-<span class="built_in">table</span> results/<span class="built_in">table</span>.qza --o-clustering results/hierarchy.qza</span><br></pre></td></tr></table></figure>
<p>gradient-clustering (the problem of overfitting, use it with caution)</p>
<figure class="highlight dsconfig"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="string">qiime </span><span class="string">gneiss </span><span class="string">gradient-clustering </span><span class="built_in">--i-table</span> <span class="string">results/</span><span class="string">table.</span><span class="string">qza </span><span class="built_in">--m-gradient-file</span> <span class="string">raw_data/</span><span class="string">mapping.</span><span class="string">txt </span><span class="built_in">--m-gradient-column</span> <span class="string">salinity </span><span class="built_in">--o-clustering</span> <span class="string">results/</span><span class="string">gradient-hierarchy.</span><span class="string">qza </span><span class="comment">#salinity</span></span><br></pre></td></tr></table></figure>
<p>visualization of <code>hierarchy.qza</code> using heatmap</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">qiime</span> <span class="comment">gneiss</span> <span class="comment">dendrogram</span><span class="literal">-</span><span class="comment">heatmap</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">table</span> <span class="comment">results/table</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">i</span><span class="literal">-</span><span class="comment">tree</span> <span class="comment">results/hierarchy</span><span class="string">.</span><span class="comment">qza</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">metadata</span><span class="literal">-</span><span class="comment">file</span> <span class="comment">raw_data/mapping</span><span class="string">.</span><span class="comment">txt</span> --<span class="comment">m</span><span class="literal">-</span><span class="comment">metadata</span><span class="literal">-</span><span class="comment">column</span> <span class="comment">group2</span> --<span class="comment">p</span><span class="literal">-</span><span class="comment">color</span><span class="literal">-</span><span class="comment">map</span> <span class="comment">seismic</span> --<span class="comment">o</span><span class="literal">-</span><span class="comment">visualization</span> <span class="comment">results/heatmap</span><span class="string">.</span><span class="comment">qzv</span></span><br></pre></td></tr></table></figure>
<p>-&gt; make different assumptions basing on the results, then do experiment to verify it.</p>
<blockquote>
<p><a href="https://docs.qiime2.org/2020.6/tutorials/gneiss/" target="_blank" rel="noopener">https://docs.qiime2.org/2020.6/tutorials/gneiss/</a></p>
</blockquote>
<h2 id="others">others</h2>
<h3 id="q2-longitudinal">q2-longitudinal</h3>
<p><a href="https://docs.qiime2.org/2020.6/tutorials/longitudinal/" target="_blank" rel="noopener">Performing longitudinal and paired sample comparisons</a></p>
<h3 id="q2-sample-classifier">q2-sample-classifier</h3>
<p><a href="https://docs.qiime2.org/2020.6/tutorials/sample-classifier/" target="_blank" rel="noopener">Predicting sample metadata values</a></p>
<h2 id="Online-visualization-tools">Online visualization tools</h2>
<p>ampvis2: <a href="https://kasperskytte.shinyapps.io/shinyampvis/" target="_blank" rel="noopener">https://kasperskytte.shinyapps.io/shinyampvis/</a><br>
needs <code>feature_table_taxonomy.biom</code> and <code>mapping.txt</code> files.</p>
<p>Ref:</p>
<blockquote>
<p>qiime2: <a href="https://qiime2.org/" target="_blank" rel="noopener">https://qiime2.org/</a><br>
微生物组16S rRNA数据分析小结: <a href="https://mp.weixin.qq.com/s/PDSk-zjKxQkTZfbc7bpH8A" target="_blank" rel="noopener">https://mp.weixin.qq.com/s/PDSk-zjKxQkTZfbc7bpH8A</a><br>
扩增子分析流程：<a href="https://mp.weixin.qq.com/s/V156q-jLhfJtFdDd_NQWjQ" target="_blank" rel="noopener">https://mp.weixin.qq.com/s/V156q-jLhfJtFdDd_NQWjQ</a></p>
</blockquote>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/qiime2/" rel="tag"># qiime2</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/07/02/biosoft-bracken/" rel="prev" title="bracken - a highly accurate statistical method that computes the abundance of species in DNA sequences from a metagenomics sample">
      <i class="fa fa-chevron-left"></i> bracken - a highly accurate statistical method that computes the abundance of species in DNA sequences from a metagenomics sample
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/07/10/16s-alphadiversity/" rel="next" title="16S_alpha diversity analysis">
      16S_alpha diversity analysis <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  

  </div>


          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">1.</span> <span class="nav-text">Environment</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">2.</span> <span class="nav-text">Data</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#"><span class="nav-number">3.</span> <span class="nav-text">Workflow</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Check-the-quality-of-raw-reads"><span class="nav-number">3.1.</span> <span class="nav-text">Check the quality of raw reads</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Importing-data"><span class="nav-number">3.2.</span> <span class="nav-text">Importing data</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Trim-adapter-optional"><span class="nav-number">3.3.</span> <span class="nav-text">Trim adapter (optional)</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Quality-control"><span class="nav-number">3.4.</span> <span class="nav-text">Quality control</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#stats-demux"><span class="nav-number">3.4.1.</span> <span class="nav-text">stats_demux</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#stats-dada2"><span class="nav-number">3.4.2.</span> <span class="nav-text">stats_dada2</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Filter-ASV-optional"><span class="nav-number">3.5.</span> <span class="nav-text">Filter ASV (optional)</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#don’t-filter"><span class="nav-number">3.5.1.</span> <span class="nav-text">don’t filter</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#filter"><span class="nav-number">3.5.2.</span> <span class="nav-text">filter</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Sequence-alignment"><span class="nav-number">3.6.</span> <span class="nav-text">Sequence alignment</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Taxonomic-assignment"><span class="nav-number">3.7.</span> <span class="nav-text">Taxonomic assignment</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Phylogenetic-tree"><span class="nav-number">3.8.</span> <span class="nav-text">Phylogenetic tree</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Diversity"><span class="nav-number">3.9.</span> <span class="nav-text">Diversity</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Alpha-diversity"><span class="nav-number">3.9.1.</span> <span class="nav-text">Alpha diversity</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#alpha-rarefaction"><span class="nav-number">3.9.1.1.</span> <span class="nav-text">alpha rarefaction</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#significance-analysis-kruskal-Wallis-test"><span class="nav-number">3.9.1.2.</span> <span class="nav-text">significance analysis (kruskal-Wallis test)</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Beta-diversity"><span class="nav-number">3.9.2.</span> <span class="nav-text">Beta diversity</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#PCoA-plot"><span class="nav-number">3.9.2.1.</span> <span class="nav-text">PCoA plot</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#significance-analysis-PERMANOVA-and-paired-comparison"><span class="nav-number">3.9.2.2.</span> <span class="nav-text">significance analysis (PERMANOVA and paired comparison)</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Abundance-table"><span class="nav-number">3.10.</span> <span class="nav-text">Abundance table</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Absolute-abundance"><span class="nav-number">3.10.1.</span> <span class="nav-text">Absolute abundance</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Relative-abundance"><span class="nav-number">3.10.2.</span> <span class="nav-text">Relative abundance</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Relative-abundance-barplot"><span class="nav-number">3.10.3.</span> <span class="nav-text">Relative abundance barplot</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Absolute-abundance-heatmap"><span class="nav-number">3.10.4.</span> <span class="nav-text">Absolute abundance heatmap</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Abundance-comparison-ANCOM"><span class="nav-number">3.10.5.</span> <span class="nav-text">Abundance comparison (ANCOM)</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Differential-abundance-analysis"><span class="nav-number">3.10.6.</span> <span class="nav-text">Differential abundance analysis</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#others"><span class="nav-number">3.11.</span> <span class="nav-text">others</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#q2-longitudinal"><span class="nav-number">3.11.1.</span> <span class="nav-text">q2-longitudinal</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#q2-sample-classifier"><span class="nav-number">3.11.2.</span> <span class="nav-text">q2-sample-classifier</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Online-visualization-tools"><span class="nav-number">3.12.</span> <span class="nav-text">Online visualization tools</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Yunmeng"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">Yunmeng</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">80</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">21</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/ItsYunmeng" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-github"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://twitter.com/ItsYunmeng" title="Twitter → https:&#x2F;&#x2F;twitter.com&#x2F;ItsYunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-twitter"></i>Twitter</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://www.instagram.com/itsyunmeng" title="Instagram → https:&#x2F;&#x2F;www.instagram.com&#x2F;itsyunmeng" rel="noopener" target="_blank"><i class="fa fa-fw fa-instagram"></i>Instagram</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yunmengli@hotmail.com" title="E-Mail → mailto:yunmengli@hotmail.com" rel="noopener" target="_blank"><i class="fa fa-fw fa-envelope"></i>E-Mail</a>
      </span>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Yunmeng</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> v4.2.0
  </div>
  <span class="post-meta-divider">|</span>
  <div class="theme-info">Theme – <a href="https://theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Gemini</a> v7.7.2
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>











<script>
if (document.querySelectorAll('pre.mermaid').length) {
  NexT.utils.getScript('//cdn.jsdelivr.net/npm/mermaid@8/dist/mermaid.min.js', () => {
    mermaid.initialize({
      theme    : 'forest',
      logLevel : 3,
      flowchart: { curve     : 'linear' },
      gantt    : { axisFormat: '%m/%d/%Y' },
      sequence : { actorMargin: 50 }
    });
  }, window.mermaid);
}
</script>


  

  

  

</body>
</html>
